<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yunqi Yang" />


<title>UKB mr result: height as exposure</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">simulation-calibrated-mr</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/yunqiyang0215/simulation-calibrated-mr">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">UKB mr result: height as exposure</h1>
<h4 class="author">Yunqi Yang</h4>
<h4 class="date">8/6/2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-08-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>simulation-calibrated-mr/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230217code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20230217)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230217code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230217)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_
</td>
<td style="text-align:left;">
data/snp_list/height_
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215simulationcalibratedmrtree4a9184a13ad7d4864192dd08d9d079d8c923f544targetblank4a9184aa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/tree/4a9184a13ad7d4864192dd08d9d079d8c923f544" target="_blank">4a9184a</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215simulationcalibratedmrtree4a9184a13ad7d4864192dd08d9d079d8c923f544targetblank4a9184aa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/tree/4a9184a13ad7d4864192dd08d9d079d8c923f544" target="_blank">4a9184a</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    scripts/.DS_Store

Untracked files:
    Untracked:  output/ukb_mr_sib1.pdf
    Untracked:  output/ukb_mr_sib2.pdf
    Untracked:  output/ukb_mr_trio1.pdf

Unstaged changes:
    Deleted:    scripts/real_data_mr_bmi.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/ukb_mr_height_202407.Rmd</code>) and HTML (<code>docs/ukb_mr_height_202407.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/4a9184a13ad7d4864192dd08d9d079d8c923f544/analysis/ukb_mr_height_202407.Rmd" target="_blank">4a9184a</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-28
</td>
<td>
wflow_publish("analysis/ukb_mr_height_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/21f897c9bcbb2f0c9762e471719b24c32b00fd6d/docs/ukb_mr_height_202407.html" target="_blank">21f897c</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/3fc513dfd97e1f089959c8bacdb2e869568a3e51/analysis/ukb_mr_height_202407.Rmd" target="_blank">3fc513d</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_height_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/899d6f24b1609850277ec85ec3c968b64ee5a134/docs/ukb_mr_height_202407.html" target="_blank">899d6f2</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/e02d50c682e649ff36b71cd84314e410f7bce16a/analysis/ukb_mr_height_202407.Rmd" target="_blank">e02d50c</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_height_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/a0fc61940e1fdd2331d5d76cba85712e7461aa05/docs/ukb_mr_height_202407.html" target="_blank">a0fc619</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/bbeb6fcdb2da2aafe28334f0bc3e52e3987e0848/analysis/ukb_mr_height_202407.Rmd" target="_blank">bbeb6fc</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
wflow_publish("analysis/ukb_mr_height_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/e7ccc3f8244c4c66e754bb93d85e633831425388/docs/ukb_mr_height_202407.html" target="_blank">e7ccc3f</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/87ce4f66b6390ae8c19620d5463def5df700225c/analysis/ukb_mr_height_202407.Rmd" target="_blank">87ce4f6</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
wflow_publish("analysis/ukb_mr_height_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/edcc3ab3c2d1a54a22e31359af89437a9d945c9b/docs/ukb_mr_height_202407.html" target="_blank">edcc3ab</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/08fc47d2a2930da24472e2579f70a1a3d7e0aeaf/analysis/ukb_mr_height_202407.Rmd" target="_blank">08fc47d</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
wflow_publish("analysis/ukb_mr_height_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/25f6ae12ec650ed8521dfd596f7d3f3586814b64/docs/ukb_mr_height_202407.html" target="_blank">25f6ae1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/9a23c2f3fdf9f15e23f4d6e704fce2d8578e0f74/analysis/ukb_mr_height_202407.Rmd" target="_blank">9a23c2f</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
wflow_publish("analysis/ukb_mr_height_202407.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="snp-selection-file-available-at" class="section level4">
<h4>SNP selection file available at:</h4>
<p><a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/master/analysis/snp_selection.R" class="uri">https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/master/analysis/snp_selection.R</a></p>
<pre class="r"><code>library(MendelianRandomization)
library(mr.raps)</code></pre>
</div>
<div id="trio-height-on-education-years" class="section level3">
<h3>1. Trio: height on education years</h3>
<pre class="r"><code>exp = &quot;height&quot;
out = &quot;education_yrs&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(3,3))
pvals = c(1e-2, 1e-5, 1e-8)

for (i in 1:length(pvals)){
  height_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% height_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% height_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: height on EduYrs: pval=&#39;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.mle: height on EduYrs: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.shrinkage: height on EduYrs: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>          calibrated         raw internal.gwas external.gwas
point.est 0.04043747 -0.03407905   -0.08509076    0.05958143
std       0.06891892  0.06790924    0.06625219    0.01052908</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est  2.3629973 -0.8542164    -0.5106406   0.062693017
std        0.5461075  0.3493194     0.5251644   0.007832986</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), :
The estimating equations might have another finite root. The closest root is
beta = 2.36299730351689 and the other root is beta = -4.3518975260533 and the
initialization is beta = 2.36299730351689</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est   2.362997 -0.8542163    -0.5106406   0.062693017
std         2.884592  2.0286276     0.3978524   0.008119601</code></pre>
<pre><code>          calibrated         raw internal.gwas external.gwas
point.est 0.08225847 -0.02056590   -0.14775606    0.05272202
std       0.09154057  0.09420451    0.09016535    0.01147496</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est  0.6538982 -0.2170825    -0.5083410    0.05429185
std        0.2924300  0.4080238     0.5599028    0.00810696</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est  0.6538981 -0.2170825    -0.5083410    0.05429185
std        0.6856364  0.9620983     0.3123013    0.00829190</code></pre>
<pre><code>          calibrated         raw internal.gwas external.gwas
point.est 0.08922501 0.008688713    -0.1935322    0.04777366
std       0.11233873 0.114751981     0.1097437    0.01217290</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est  0.4576931 0.04300039    -0.4923192   0.048739701
std        0.3006106 0.65352861     0.5230398   0.008556461</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est  0.4576930 0.04300037    -0.4923192   0.048739701
std        0.5471996 0.56807049     0.2816242   0.008690088</code></pre>
<p><img src="figure/ukb_mr_height_202407.Rmd/unnamed-chunk-5-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/899d6f24b1609850277ec85ec3c968b64ee5a134/docs/figure/ukb_mr_height_202407.Rmd/unnamed-chunk-5-1.png" target="_blank">899d6f2</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sibling-height-on-education-years" class="section level3">
<h3>2. Sibling: height on education years</h3>
<pre class="r"><code>exp = &quot;height&quot;
out = &quot;education_yrs&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(3,3))
pvals = c(1e-2, 1e-5, 1e-8)

for (i in 1:length(pvals)){
  height_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% height_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% height_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: height on EduYrs: pval=&#39;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.mle: height on EduYrs: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.shrinkage: height on EduYrs: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.03354192 0.02265064    0.04180293    0.05958143
std       0.03753601 0.03891857    0.03030769    0.01052908</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.05197435 0.03527708    0.06128137   0.062693017
std       0.05320521 0.05995801    0.04214046   0.007832986</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.05197436 0.03527706    0.06128138   0.062693017
std       0.05622927 0.06035037    0.04348188   0.008119601</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.01597153 0.01001812    0.05570258    0.05272202
std       0.04376749 0.04510227    0.03379875    0.01147496</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.02095027 0.01312213    0.06910086    0.05429185
std       0.05093107 0.05643735    0.04100659    0.00810696</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.02095028 0.01312212    0.06910085    0.05429185
std       0.05387945 0.05767155    0.04138274    0.00829190</code></pre>
<pre><code>            calibrated          raw internal.gwas external.gwas
point.est -0.001254538 -0.005053235    0.03165832    0.04777366
std        0.046293175  0.048291601    0.03670774    0.01217290</code></pre>
<pre><code>            calibrated          raw internal.gwas external.gwas
point.est -0.001497973 -0.006048166    0.03680046   0.048739701
std        0.052896695  0.058012632    0.04147073   0.008556461</code></pre>
<pre><code>            calibrated          raw internal.gwas external.gwas
point.est -0.001497983 -0.006048166    0.03680046   0.048739701
std        0.053725968  0.057799960    0.04183783   0.008690088</code></pre>
<p><img src="figure/ukb_mr_height_202407.Rmd/unnamed-chunk-9-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/899d6f24b1609850277ec85ec3c968b64ee5a134/docs/figure/ukb_mr_height_202407.Rmd/unnamed-chunk-9-1.png" target="_blank">899d6f2</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/e7ccc3f8244c4c66e754bb93d85e633831425388/docs/figure/ukb_mr_height_202407.Rmd/unnamed-chunk-9-1.png" target="_blank">e7ccc3f</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/25f6ae12ec650ed8521dfd596f7d3f3586814b64/docs/figure/ukb_mr_height_202407.Rmd/unnamed-chunk-9-1.png" target="_blank">25f6ae1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="trio-height-on-smoking" class="section level3">
<h3>3. Trio: height on smoking</h3>
<pre class="r"><code>exp = &quot;height&quot;
out = &quot;smoking&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(3,3))
pvals = c(1e-2, 1e-5, 1e-8)

for (i in 1:length(pvals)){
  height_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% height_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% height_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: height on smoking: pval=&#39;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.mle: height on smoking: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.shrinkage: height on smoking: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>            calibrated         raw internal.gwas external.gwas
point.est -0.005447051 0.004001921  -0.007324626  -0.027220993
std        0.041117068 0.042066045   0.041909529   0.008164276</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est -0.6723959 0.07667887   -0.04938297  -0.028296793
std        0.2267766 0.35355213    0.28505007   0.007041556</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= -0.672395844386822 and the other root is beta = 0.00709065090949905 and the
initialization is beta = -0.672395867506946</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est -0.6723958 0.07667889    -0.0493830  -0.028296793
std        2.4012447 0.79930942     0.2825657   0.007160744</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.05812646 -0.04123792    0.01040347  -0.026124598
std        0.05428893  0.05803850    0.05701364   0.008888434</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est -0.2432089 -0.3125719    0.04262647   -0.02671950
std        0.2571208  0.3434869    0.27446221    0.00726963</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est -0.2432089 -0.3125719    0.04262646    -0.0267195
std        0.2246204  0.4310468    0.23363177     0.0073537</code></pre>
<pre><code>           calibrated        raw internal.gwas external.gwas
point.est -0.05818022 0.04388698   0.004902194  -0.026914696
std        0.06733095 0.07337916   0.069466596   0.009356206</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est -0.1612891 0.1747398    0.01622216   -0.02733302
std        0.3547175 0.7388039    0.23954681    0.00766571</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est -0.1612892 0.1747398    0.01622215  -0.027333025
std        0.1867844 0.2947822    0.22987596   0.007723991</code></pre>
<p><img src="figure/ukb_mr_height_202407.Rmd/unnamed-chunk-13-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/899d6f24b1609850277ec85ec3c968b64ee5a134/docs/figure/ukb_mr_height_202407.Rmd/unnamed-chunk-13-1.png" target="_blank">899d6f2</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sibling-height-on-smoking" class="section level3">
<h3>4. Sibling: height on smoking</h3>
<pre class="r"><code>exp = &quot;height&quot;
out = &quot;smoking&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(3,3))
pvals = c(1e-2, 1e-5, 1e-8)

for (i in 1:length(pvals)){
  height_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% height_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% height_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: height on smoking: pval=&#39;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.mle: height on smoking: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.shrinkage: height on smoking: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.11733499 0.13424534  -0.003478644  -0.027220993
std       0.03266814 0.03593168   0.024149375   0.008164276</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.17758948 0.20797857  -0.004978843  -0.028296793
std       0.04828352 0.05585815   0.034835707   0.007041556</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.17758948 0.20797858  -0.004978836  -0.028296793
std       0.04901748 0.05571221   0.034564041   0.007160744</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.13358333 0.15399043   0.006290623  -0.026124598
std       0.03694192 0.04075705   0.027888018   0.008888434</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.16965507 0.19834567   0.007852882   -0.02671950
std       0.04639631 0.05296188   0.033246115    0.00726963</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.16965508 0.19834567   0.007852887    -0.0267195
std       0.04660939 0.05268355   0.033897783     0.0073537</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.12799657 0.15581892    0.01518027  -0.026914696
std       0.04053003 0.04472936    0.02930472   0.009356206</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.15165532 0.18575462    0.01743306   -0.02733302
std       0.04747186 0.05401673    0.03399759    0.00766571</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.15165531 0.18575461    0.01743306  -0.027333025
std       0.04762679 0.05350238    0.03365592   0.007723991</code></pre>
<p><img src="figure/ukb_mr_height_202407.Rmd/unnamed-chunk-17-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/899d6f24b1609850277ec85ec3c968b64ee5a134/docs/figure/ukb_mr_height_202407.Rmd/unnamed-chunk-17-1.png" target="_blank">899d6f2</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="trio-height-on-diabetes" class="section level3">
<h3>5. Trio: height on diabetes</h3>
<pre class="r"><code>exp = &quot;height&quot;
out = &quot;diabetes&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(3,3))
pvals = c(1e-2, 1e-5, 1e-8)

for (i in 1:length(pvals)){
  height_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% height_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% height_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: height on diabetes: pval=&#39;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.mle: height on diabetes: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.shrinkage: height on diabetes: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.07166396 -0.01047502   -0.01649322  -0.010400585
std        0.03936082  0.03610097    0.04011655   0.008825948</code></pre>
<pre><code>          calibrated         raw internal.gwas external.gwas
point.est -1.2845002 -0.06805657    -0.1170023  -0.010895007
std        0.3027953  0.25287926     0.2583183   0.006908958</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= -1.28450023092075 and the other root is beta = 0.0998168464845803 and the
initialization is beta = -1.28450022379505</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= -1.28450023092075 and the other root is beta = 1.33963784862914 and the
initialization is beta = -1.28450022379505</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), :
The estimating equations might have another finite root. The closest root is
beta = -1.28450023092075 and the other root is beta = 9.7008135475188 and the
initialization is beta = -1.28450022379505</code></pre>
<pre><code>          calibrated         raw internal.gwas external.gwas
point.est  -1.284500 -0.06805655    -0.1170023  -0.010895010
std         1.266853  0.21894962     0.2835957   0.007110696</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.05803812 -0.02694655   -0.03093218  -0.007298264
std        0.05151326  0.05096955    0.05563857   0.009396712</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est -0.6104849 -0.1329904    -0.1625104  -0.007477625
std        0.1828465  0.1617767     0.2145497   0.007149181</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est -0.6104848 -0.1329904    -0.1625104  -0.007477627
std        0.5656641  0.2190425     0.2861089   0.007268736</code></pre>
<pre><code>           calibrated        raw internal.gwas external.gwas
point.est -0.10594584 0.01459922    0.02132582   -0.01163219
std        0.06100819 0.06440659    0.06627347    0.01079683</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est -0.3860901 0.0472642    0.06432806  -0.011865935
std        0.1518004 0.1751766    0.25086817   0.007530539</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est -0.3860901 0.04726421    0.06432804  -0.011865935
std        0.2908897 0.20618750    0.19996236   0.007651697</code></pre>
<p><img src="figure/ukb_mr_height_202407.Rmd/unnamed-chunk-21-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="sibling-height-on-diabetes" class="section level3">
<h3>6. Sibling: height on diabetes</h3>
<pre class="r"><code>exp = &quot;height&quot;
out = &quot;diabetes&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(3,3))
pvals = c(1e-2, 1e-5, 1e-8)

for (i in 1:length(pvals)){
  height_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% height_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% height_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: height on diabetes: pval=&#39;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.mle: height on diabetes: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.shrinkage: height on diabetes: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>           calibrated        raw internal.gwas external.gwas
point.est 0.006509713 0.06573522    0.05449189  -0.010400585
std       0.031633098 0.03524283    0.02478224   0.008825948</code></pre>
<pre><code>           calibrated        raw internal.gwas external.gwas
point.est 0.009789818 0.10272215    0.08197984  -0.010895007
std       0.047179037 0.05409378    0.03297427   0.006908958</code></pre>
<pre><code>           calibrated        raw internal.gwas external.gwas
point.est 0.009789818 0.10272214    0.08197985  -0.010895010
std       0.047411298 0.05465259    0.03551436   0.007110696</code></pre>
<pre><code>           calibrated        raw internal.gwas external.gwas
point.est 0.009078447 0.04601616    0.06742111  -0.007298264
std       0.036337384 0.03985830    0.02734345   0.009396712</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.01157426 0.05943460    0.08434153  -0.007477625
std       0.04504194 0.05143906    0.03244113   0.007149181</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est 0.01157426 0.0594346    0.08434153  -0.007477627
std       0.04555426 0.0513751    0.03320025   0.007268736</code></pre>
<pre><code>            calibrated        raw internal.gwas external.gwas
point.est -0.002371786 0.04229345    0.07142877   -0.01163219
std        0.039168711 0.04360443    0.03053450    0.01079683</code></pre>
<pre><code>            calibrated        raw internal.gwas external.gwas
point.est -0.002799403 0.05042485    0.08433139  -0.011865935
std        0.046274942 0.05249136    0.03271280   0.007530539</code></pre>
<pre><code>           calibrated        raw internal.gwas external.gwas
point.est -0.00279941 0.05042484    0.08433138  -0.011865935
std        0.04618451 0.05200048    0.03381664   0.007651697</code></pre>
<p><img src="figure/ukb_mr_height_202407.Rmd/unnamed-chunk-25-1.png" width="1440" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin20.6.0 (64-bit)
Running under: macOS Monterey 12.0.1

Matrix products: default
BLAS:   /usr/local/Cellar/openblas/0.3.18/lib/libopenblasp-r0.3.18.dylib
LAPACK: /usr/local/Cellar/r/4.1.1_1/lib/R/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] mr.raps_0.4.1                MendelianRandomization_0.7.0
[3] workflowr_1.6.2             

loaded via a namespace (and not attached):
 [1] httr_1.4.6         sass_0.4.4         tidyr_1.2.1        jsonlite_1.7.2    
 [5] viridisLite_0.4.0  splines_4.1.1      foreach_1.5.2      bslib_0.4.1       
 [9] highr_0.9          arrangements_1.1.9 yaml_2.2.1         robustbase_0.95-1 
[13] ggrepel_0.9.3      pillar_1.9.0       lattice_0.20-44    quantreg_5.94     
[17] glue_1.4.2         digest_0.6.28      promises_1.2.0.1   iterpc_0.4.2      
[21] colorspace_2.0-2   plyr_1.8.6         htmltools_0.5.5    httpuv_1.6.3      
[25] Matrix_1.5-3       pkgconfig_2.0.3    httpcode_0.3.0     SparseM_1.81      
[29] purrr_0.3.4        scales_1.2.1       rsnps_0.5.0.0      whisker_0.4       
[33] rootSolve_1.8.2.3  later_1.3.0        MatrixModels_0.5-1 git2r_0.28.0      
[37] tibble_3.1.5       gmp_0.7-1          generics_0.1.2     ggplot2_3.4.3     
[41] ellipsis_0.3.2     cachem_1.0.6       lazyeval_0.2.2     cli_3.6.1         
[45] survival_3.2-11    magrittr_2.0.1     evaluate_0.14      fs_1.5.0          
[49] fansi_0.5.0        MASS_7.3-54        tools_4.1.1        data.table_1.14.9 
[53] lifecycle_1.0.3    stringr_1.4.0      plotly_4.10.1      munsell_0.5.0     
[57] glmnet_4.1-4       compiler_4.1.1     jquerylib_0.1.4    rlang_1.1.1       
[61] grid_4.1.1         iterators_1.0.14   rstudioapi_0.13    rjson_0.2.21      
[65] htmlwidgets_1.6.2  rmarkdown_2.11     gtable_0.3.0       codetools_0.2-18  
[69] curl_4.3.2         R6_2.5.1           gridExtra_2.3      knitr_1.36        
[73] dplyr_1.0.7        fastmap_1.1.0      utf8_1.2.2         nortest_1.0-4     
[77] rprojroot_2.0.2    shape_1.4.6        stringi_1.7.5      crul_1.4.0        
[81] Rcpp_1.0.8.3       vctrs_0.6.3        DEoptimR_1.0-13    tidyselect_1.2.0  
[85] xfun_0.27         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
