<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yunqi Yang" />


<title>UKB mr result: BMI as exposure</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">simulation-calibrated-mr</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/yunqiyang0215/simulation-calibrated-mr">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">UKB mr result: BMI as exposure</h1>
<h4 class="author">Yunqi Yang</h4>
<h4 class="date">8/6/2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-08-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>simulation-calibrated-mr/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230217code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20230217)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230217code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230217)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_
</td>
<td style="text-align:left;">
data/snp_list/bmi_
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215simulationcalibratedmrtree3abf2d9cca5957acacab0284a575f2d5ae9b2df7targetblank3abf2d9a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/tree/3abf2d9cca5957acacab0284a575f2d5ae9b2df7" target="_blank">3abf2d9</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215simulationcalibratedmrtree3abf2d9cca5957acacab0284a575f2d5ae9b2df7targetblank3abf2d9a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/tree/3abf2d9cca5957acacab0284a575f2d5ae9b2df7" target="_blank">3abf2d9</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    scripts/.DS_Store

Untracked files:
    Untracked:  output/ukb_mr_sib1.pdf
    Untracked:  output/ukb_mr_sib2.pdf
    Untracked:  output/ukb_mr_trio1.pdf

Unstaged changes:
    Modified:   output/real_data_calibration.pdf
    Deleted:    scripts/real_data_mr_bmi.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/ukb_mr_bmi_202407.Rmd</code>) and HTML (<code>docs/ukb_mr_bmi_202407.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/3abf2d9cca5957acacab0284a575f2d5ae9b2df7/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">3abf2d9</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-29
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/1a0e15fc3551d27e4e5c4b56783188236084c9e4/docs/ukb_mr_bmi_202407.html" target="_blank">1a0e15f</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/6102007b3125c79e497cb80074c597403fab099f/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">6102007</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-28
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/bd881a1d96f590e4a173da83ebe4e7347a402f37/docs/ukb_mr_bmi_202407.html" target="_blank">bd881a1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/d1323c0bb3b797bfd051b597066aeb7ab0b65af1/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">d1323c0</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-13
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/2919327adb86772679cdfb5019f162dfaae45916/docs/ukb_mr_bmi_202407.html" target="_blank">2919327</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/40643d5cdf22f2e10612f64d5a0ed90c895dd147/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">40643d5</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/ceb7aae41766ff85e99caf85621cbf766ab389a0/docs/ukb_mr_bmi_202407.html" target="_blank">ceb7aae</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/c2285a525f132517309bbc2c2f14a85ef170728e/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">c2285a5</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/47e777a9d3974dcd94c63cfeac246bca3225ba56/docs/ukb_mr_bmi_202407.html" target="_blank">47e777a</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/baa03061f7734cc1f1c811351edbac75670aecc4/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">baa0306</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/711b69207310a97738a72fe4d53a67b9e02cae57/docs/ukb_mr_bmi_202407.html" target="_blank">711b692</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/2ab18942089b73b65482166b0826fff8fd1a7ccb/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">2ab1894</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/818190488ce7da0f341fa7b5807a7f8817a01c6f/docs/ukb_mr_bmi_202407.html" target="_blank">8181904</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/981aeb1eea621c7190c904be5a4e9e0eb5d1610a/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">981aeb1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/d17ce9773a200d70f36a81c6e4f8158ba5603455/docs/ukb_mr_bmi_202407.html" target="_blank">d17ce97</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/ce497d3f699eb69d0a7ecd3862567282d298b7ef/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">ce497d3</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/b26f07491f05dc3cdf48e643fa4b7c69966fcfce/docs/ukb_mr_bmi_202407.html" target="_blank">b26f074</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/e2b5caca6796af8957601a6289a31eace7d69c44/docs/ukb_mr_bmi_202407.html" target="_blank">e2b5cac</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/e3f8ab9d4abdf67232857b4311bfd9e2b4a3f9a1/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">e3f8ab9</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/2b0c94c3a51580c5ce73113863de7dcdd685fbc7/docs/ukb_mr_bmi_202407.html" target="_blank">2b0c94c</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/c2819fa100af91266389c8e5cd71a7072e838c86/docs/ukb_mr_bmi_202407.html" target="_blank">c2819fa</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/f53447f0e212bf11f19c1f9bf689ca6c89e5a724/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">f53447f</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/fe15e0731151212f52ec1d0c9b3c09a7cd8c589c/docs/ukb_mr_bmi_202407.html" target="_blank">fe15e07</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/0dc3b0bae18b0269c113ecb92ad4aad75d91b55b/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">0dc3b0b</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/1584f2809565e293474f8d371a271cb50a587041/docs/ukb_mr_bmi_202407.html" target="_blank">1584f28</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/8958ff17ef2ae2da9a9a0889d2d4b41c6f3f2c94/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">8958ff1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="snp-selection-file-available-at" class="section level4">
<h4>SNP selection file available at:</h4>
<p><a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/master/analysis/snp_selection.R" class="uri">https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/master/analysis/snp_selection.R</a></p>
<pre class="r"><code>library(MendelianRandomization)
library(mr.raps)</code></pre>
</div>
<div id="trio-bmi-on-diabetes" class="section level3">
<h3>1. Trio: BMI on diabetes</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;diabetes&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(2,4))
pvals = c(1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # debiased IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_divw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_divw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_divw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_divw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;Debiased IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.mle: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.shrinkage: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.19199294 0.14767053    0.16415084     0.2136415
std       0.07049309 0.07261298    0.07206222     0.0295015</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.7034634 0.8917999      1.794472    0.21935251
std        0.3580443 0.6988244      2.104226    0.03041011</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.6377509 0.9228341     0.6800884    0.25244038
std        0.2189662 0.5609100     1.8210034    0.01181858</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.637750917841163 and the other root is beta = -3.24017610354703 and the
initialization is beta = 0.637750935727596</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.637750917841163 and the other root is beta = -3.2401792558122 and the
initialization is beta = 0.637750935727596</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.922834055389757 and the other root is beta =
-1.44220910428254 and the initialization is beta = 0.922834055389757</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.6377509 0.9228341     0.6800884    0.25244038
std        0.2982836 0.4323314     0.3151571    0.01432002</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.2952433 0.1375211     0.1432149    0.21792086
std        0.1138949 0.1110828     0.1144336    0.05425065</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.7451782 0.4178795      0.563110    0.22090121
std        0.4135047 0.3845947      0.560563    0.05489029</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.6510441 0.5372329      0.325047    0.26155270
std        0.2402355 0.3406265      1.013362    0.01463554</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.651044138967306 and the other root is beta = -1.93946468772688 and the
initialization is beta = 0.651044138967306</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.651044138967306 and the other root is beta = -1.93946468704843 and the
initialization is beta = 0.651044138967306</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.537232941299394 and the other root is beta =
-2.22445831089624 and the initialization is beta = 0.537232941299394</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.6510441 0.5372329     0.3250470    0.26155270
std        0.3190874 0.4287565     0.2658774    0.01838022</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-5-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/47e777a9d3974dcd94c63cfeac246bca3225ba56/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-5-1.png" target="_blank">47e777a</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/711b69207310a97738a72fe4d53a67b9e02cae57/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-5-1.png" target="_blank">711b692</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/d17ce9773a200d70f36a81c6e4f8158ba5603455/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-5-1.png" target="_blank">d17ce97</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sibling-bmi-on-diabetes" class="section level3">
<h3>2. Sibling: BMI on diabetes</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;diabetes&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(2,4))
pvals = c(1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # debiased IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_divw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_divw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_divw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_divw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;Debiased IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.mle: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.shrinkage: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.21049994 0.18066066    0.18289797     0.2136415
std       0.05647031 0.05935615    0.04844138     0.0295015</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.32237072 0.29611540    0.26670543    0.21935251
std       0.08823803 0.09920197    0.07178127    0.03041011</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.38186359 0.34226803    0.29933464    0.25244038
std       0.07083702 0.08267482    0.05999259    0.01181858</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est 0.38186358 0.3422680    0.29933464    0.25244038
std       0.08976942 0.1016511    0.07103329    0.01432002</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.33130600 0.34944851    0.22314574    0.21792086
std       0.08999924 0.09149393    0.08083187    0.05425065</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est   0.418180 0.4683126     0.2770039    0.22090121
std         0.114872 0.1255145     0.1006931    0.05489029</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est 0.50412307 0.5189957    0.33804200    0.26155270
std       0.08335365 0.1035925    0.06983543    0.01463554</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.504123068547964 and the other root is beta = -2.33412789232438 and the
initialization is beta = 0.504123073461214</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.504123068547964 and the other root is beta = -2.33412783377308 and the
initialization is beta = 0.504123073461214</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.504123068547964 and the other root is beta = -2.33412796902626 and the
initialization is beta = 0.504123073461214</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var),
sqrt(dat2$int.var)): The estimating equations might have another finite root.
The closest root is beta = 0.338042000110738 and the other root is beta =
-2.92868159205393 and the initialization is beta = 0.33804200458043</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var),
sqrt(dat2$int.var)): The estimating equations might have another finite root.
The closest root is beta = 0.338042000110738 and the other root is beta =
-2.92868231451648 and the initialization is beta = 0.33804200458043</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.5041231 0.5189957    0.33804200    0.26155270
std        0.1057312 0.1181563    0.09166537    0.01838022</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-9-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/bd881a1d96f590e4a173da83ebe4e7347a402f37/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-9-1.png" target="_blank">bd881a1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/47e777a9d3974dcd94c63cfeac246bca3225ba56/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-9-1.png" target="_blank">47e777a</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/711b69207310a97738a72fe4d53a67b9e02cae57/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-9-1.png" target="_blank">711b692</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/d17ce9773a200d70f36a81c6e4f8158ba5603455/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-9-1.png" target="_blank">d17ce97</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/e2b5caca6796af8957601a6289a31eace7d69c44/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-9-1.png" target="_blank">e2b5cac</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/c2819fa100af91266389c8e5cd71a7072e838c86/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-9-1.png" target="_blank">c2819fa</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/1584f2809565e293474f8d371a271cb50a587041/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-9-1.png" target="_blank">1584f28</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="trio-bmi-on-education-years" class="section level3">
<h3>3. Trio: BMI on education years</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;education_yrs&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(2,4))
pvals = c(1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # debiased IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_divw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_divw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_divw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_divw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;Debiased IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.mle: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.shrinkage: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.2298938 0.1893570   -0.01709613   -0.12169550
std        0.1014469 0.1129532    0.11970946    0.02253355</code></pre>
<pre><code>          calibrated      raw internal.gwas external.gwas
point.est  0.7814661 1.312071    -0.2028969    -0.1249487
std        0.4108790 1.169720     1.4409914     0.0232032</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.5413199 0.7430371     -0.150400   -0.13133694
std        0.5093184 1.2533918      2.102216    0.01351656</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.541319940126336 and the other root is beta = -4.86989528926554 and the
initialization is beta = 0.541319940126336</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.743037052466926 and the other root is beta =
-4.13456501961995 and the initialization is beta = 0.743037052466926</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.743037052466926 and the other root is beta =
-4.1345650231491 and the initialization is beta = 0.743037052466926</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.743037052466926 and the other root is beta =
-4.13456503021181 and the initialization is beta = 0.743037052466926</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.5413199 0.7430371    -0.1503999    -0.1313369
std        0.2447361 0.4627413     1.0553844     0.0145233</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.1222599 0.1339975    0.05738363   -0.08705489
std        0.1627665 0.1810404    0.19745066    0.03096505</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.3021064 0.4850437     0.2398568   -0.08824547
std        0.4136838 0.7043269     0.8296594    0.03128976</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.2104791 0.2852621     0.2960719   -0.09119262
std        0.6323016 1.6475888     0.6967661    0.01690865</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.2104791 0.2852621     0.2960719   -0.09119262
std        0.2815900 0.3890766     0.9749314    0.01772449</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-13-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/bd881a1d96f590e4a173da83ebe4e7347a402f37/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-13-1.png" target="_blank">bd881a1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/47e777a9d3974dcd94c63cfeac246bca3225ba56/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-13-1.png" target="_blank">47e777a</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/711b69207310a97738a72fe4d53a67b9e02cae57/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-13-1.png" target="_blank">711b692</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/d17ce9773a200d70f36a81c6e4f8158ba5603455/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-13-1.png" target="_blank">d17ce97</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sibling-bmi-on-education-years" class="section level3">
<h3>4. Sibling: BMI on education years</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;education_yrs&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(2,4))
pvals = c(1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # debiased IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_divw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_divw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_divw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_divw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;Debiased IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.mle: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.shrinkage: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.01825871 -0.03786022   -0.20781912   -0.12169550
std        0.05513444  0.05826400    0.05312938    0.02253355</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.02793894 -0.06197481   -0.30337355    -0.1249487
std        0.08439401  0.09554138    0.08049438     0.0232032</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.02591777 -0.05723418    -0.2969445   -0.13133694
std        0.08913997  0.10134807     0.0800222    0.01351656</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.02591777 -0.05723418   -0.29694452    -0.1313369
std        0.07827802  0.08814827    0.07741541     0.0145233</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.02791726 -0.05763980   -0.12027563   -0.08705489
std        0.07464877  0.08238804    0.07331826    0.03096505</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.03515341 -0.07703227   -0.14947699   -0.08824547
std        0.09404321  0.11037339    0.09189099    0.03128976</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.03388451 -0.07216306   -0.14327112   -0.09119262
std        0.09620649  0.11514805    0.09396632    0.01690865</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.03388451 -0.07216306   -0.14327112   -0.09119262
std        0.09064139  0.10329808    0.08783489    0.01772449</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-17-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/bd881a1d96f590e4a173da83ebe4e7347a402f37/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-17-1.png" target="_blank">bd881a1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/47e777a9d3974dcd94c63cfeac246bca3225ba56/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-17-1.png" target="_blank">47e777a</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/711b69207310a97738a72fe4d53a67b9e02cae57/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-17-1.png" target="_blank">711b692</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/d17ce9773a200d70f36a81c6e4f8158ba5603455/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-17-1.png" target="_blank">d17ce97</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="trio-bmi-on-smoking" class="section level3">
<h3>5. Trio: BMI on smoking</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;smoking&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(2,4))
pvals = c(1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # debiased IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_divw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_divw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_divw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_divw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;Debiased IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.mle: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.shrinkage: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>           calibrated        raw internal.gwas external.gwas
point.est -0.07064455 0.07046356   -0.00335690    0.09231808
std        0.06517653 0.07512403    0.07547593    0.01650998</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est -0.2485183 0.4803059   -0.03635117    0.09478592
std        0.2395638 0.5929093    0.81824974    0.01696604</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est -0.2603259 0.6172662   -0.03188522    0.09708124
std        0.1940594 0.4531537    0.94481116    0.01219272</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est -0.2603259 0.6172662   -0.03188524    0.09708124
std        0.2623595 0.6230411    0.71674875    0.01263152</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est 0.08032099 0.1656224   -0.06955929    0.06982057
std       0.10853475 0.1205187    0.11984137    0.02554330</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.1929415 0.6315335    -0.2747094    0.07077546
std        0.2636111 0.5755785     0.5008755    0.02578813</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.2027309 0.5031254    -0.2219370    0.07256776
std        0.2372671 0.4569862     0.5907086    0.01514555</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.503125377566961 and the other root is beta =
-2.61255850242802 and the initialization is beta = 0.503125377566961</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.2027309 0.5031254    -0.2219370    0.07256776
std        0.2671649 0.3825085     0.3847335    0.01571885</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-21-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/bd881a1d96f590e4a173da83ebe4e7347a402f37/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-21-1.png" target="_blank">bd881a1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/ceb7aae41766ff85e99caf85621cbf766ab389a0/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-21-1.png" target="_blank">ceb7aae</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/47e777a9d3974dcd94c63cfeac246bca3225ba56/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-21-1.png" target="_blank">47e777a</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/711b69207310a97738a72fe4d53a67b9e02cae57/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-21-1.png" target="_blank">711b692</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/d17ce9773a200d70f36a81c6e4f8158ba5603455/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-21-1.png" target="_blank">d17ce97</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sibling-bmi-on-smoking" class="section level3">
<h3>6. Sibling: BMI on smoking</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;smoking&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(2,4))
pvals = c(1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # debiased IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_divw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_divw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_divw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_divw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;Debiased IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.mle: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.shrinkage: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>          calibrated         raw internal.gwas external.gwas
point.est 0.01627286 -0.02166770    0.10226795    0.09231808
std       0.05156174  0.05475607    0.04762585    0.01650998</code></pre>
<pre><code>          calibrated         raw internal.gwas external.gwas
point.est 0.02488886 -0.03538100    0.14959100    0.09478592
std       0.07869290  0.08920489    0.06990396    0.01696604</code></pre>
<pre><code>          calibrated         raw internal.gwas external.gwas
point.est 0.02621466 -0.03598158    0.16310431    0.09708124
std       0.07263472  0.08696831    0.06087014    0.01219272</code></pre>
<pre><code>          calibrated         raw internal.gwas external.gwas
point.est 0.02621466 -0.03598157    0.16310431    0.09708124
std       0.07903666  0.08950178    0.06971466    0.01263152</code></pre>
<pre><code>          calibrated          raw internal.gwas external.gwas
point.est 0.03078754 -0.005483624    0.02498641    0.06982057
std       0.07848377  0.095001021    0.06766562    0.02554330</code></pre>
<pre><code>          calibrated          raw internal.gwas external.gwas
point.est 0.03876168 -0.007314021    0.03109123    0.07077546
std       0.09839191  0.126432669    0.08351583    0.02578813</code></pre>
<pre><code>          calibrated          raw internal.gwas external.gwas
point.est 0.04271851 -0.008889594    0.03305786    0.07256776
std       0.07925882  0.091617541    0.07214723    0.01514555</code></pre>
<pre><code>          calibrated          raw internal.gwas external.gwas
point.est 0.04271851 -0.008889599    0.03305786    0.07256776
std       0.09202153  0.123183937    0.07939483    0.01571885</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-25-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/bd881a1d96f590e4a173da83ebe4e7347a402f37/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-25-1.png" target="_blank">bd881a1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/ceb7aae41766ff85e99caf85621cbf766ab389a0/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-25-1.png" target="_blank">ceb7aae</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/47e777a9d3974dcd94c63cfeac246bca3225ba56/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-25-1.png" target="_blank">47e777a</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/711b69207310a97738a72fe4d53a67b9e02cae57/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-25-1.png" target="_blank">711b692</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/818190488ce7da0f341fa7b5807a7f8817a01c6f/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-25-1.png" target="_blank">8181904</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/d17ce9773a200d70f36a81c6e4f8158ba5603455/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-25-1.png" target="_blank">d17ce97</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="trio-bmi-on-sbp" class="section level3">
<h3>7. Trio: BMI on sbp</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;sbp&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(2,4))
pvals = c(1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # debiased IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_divw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_divw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_divw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_divw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;Debiased IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.mle: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.shrinkage: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est 0.21138978 0.2931367    0.24892692    0.09569103
std       0.06285211 0.0674167    0.06765445    0.02216957</code></pre>
<pre><code>          calibrated      raw internal.gwas external.gwas
point.est  0.7719518 2.083977      2.720873    0.09824904
std        0.3314506 1.495769      3.049372    0.02282594</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.5656380 0.8250697     0.8898675    0.10573635
std        0.2802898 0.5717170     0.8238892    0.01149347</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.56563796178669 and the other root is beta = -1.89318231656941 and the
initialization is beta = 0.56563796178669</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.56563796178669 and the other root is beta = -1.89318228363097 and the
initialization is beta = 0.56563796178669</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.825069650563325 and the other root is beta =
-1.18190801141113 and the initialization is beta = 0.825069650563325</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.5656380 0.8250697     0.8898675    0.10573634
std        0.1754874 0.2116682     0.2711387    0.01279374</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.1688944 0.3006873     0.1538126    0.08211969
std        0.1122970 0.1146799     0.1075151    0.03333277</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.4321048 1.1221270     0.6105282    0.08324278
std        0.3112063 0.7432998     0.5625420    0.03369098</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.5447377 0.7574622     0.4342374    0.08823500
std        0.2472522 0.4230426     0.5580479    0.01438633</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.544737710396645 and the other root is beta = -2.18054464694072 and the
initialization is beta = 0.544737710396645</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.544737710396645 and the other root is beta = -2.18054619346576 and the
initialization is beta = 0.544737710396645</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.544737710396645 and the other root is beta = -2.18054616457695 and the
initialization is beta = 0.544737710396645</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.757462170005286 and the other root is beta =
-1.3177432385109 and the initialization is beta = 0.757462170005286</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.757462170005286 and the other root is beta =
-1.31774326397176 and the initialization is beta = 0.757462170005286</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.757462170005286 and the other root is beta =
-1.31774326496231 and the initialization is beta = 0.757462170005286</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var),
sqrt(dat2$int.var)): The estimating equations might have another finite root.
The closest root is beta = 0.434237428971582 and the other root is beta =
-2.06205171760947 and the initialization is beta = 0.434237428971582</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.5447377 0.7574622     0.4342374     0.0882350
std        0.3392008 0.3130601     0.3159413     0.0156692</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-29-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/d17ce9773a200d70f36a81c6e4f8158ba5603455/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-29-1.png" target="_blank">d17ce97</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/fe15e0731151212f52ec1d0c9b3c09a7cd8c589c/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-29-1.png" target="_blank">fe15e07</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sibling-bmi-on-sbp" class="section level3">
<h3>8. Sibling: BMI on sbp</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;sbp&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(2,4))
pvals = c(1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # debiased IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_divw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_divw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_divw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_divw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;Debiased IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.mle: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.shrinkage: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>           calibrated        raw internal.gwas external.gwas
point.est 0.008038825 0.03895568    0.13169264    0.09569103
std       0.047577104 0.05093788    0.04604206    0.02216957</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.01228596 0.06367178    0.19231279    0.09824904
std       0.07251569 0.08345705    0.06775748    0.02282594</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.01241506 0.06212765    0.20538746    0.10573635
std       0.07130883 0.08485296    0.06059521    0.01149347</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.01241506 0.06212765     0.2053875    0.10573634
std       0.07254587 0.08128627     0.0670969    0.01279374</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.03240193 0.05198779    0.09731632    0.08211969
std       0.06471221 0.07252114    0.06371926    0.03333277</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.04076926 0.06936888     0.1209837    0.08324278
std       0.08075653 0.09601657     0.0787601    0.03369098</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.04105794 0.06936308     0.1254254    0.08823500
std       0.07942956 0.09586916     0.0718071    0.01438633</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.04105794 0.06936309    0.12542535     0.0882350
std       0.08031545 0.09582484    0.07593946     0.0156692</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-33-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="trio-bmi-on-dbp" class="section level3">
<h3>9. Trio: BMI on dbp</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;dbp&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(2,4))
pvals = c(1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # debiased IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_divw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_divw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_divw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_divw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;Debiased IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.mle: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.shrinkage: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.37467980 0.48223661    0.37066438    0.16282333
std       0.06678159 0.07384397    0.07529188    0.02328566</code></pre>
<pre><code>          calibrated      raw internal.gwas external.gwas
point.est  1.3192353 3.639105      4.045134     0.1671759
std        0.4560663 2.694501      4.467368     0.0239826</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.7486043 0.9817650     1.0163779    0.17968008
std        0.3102098 0.5794792     0.9020499    0.01218798</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.981764955373023 and the other root is beta =
-1.16247625537537 and the initialization is beta = 0.981764955373023</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.981764955373023 and the other root is beta =
-1.16247626025656 and the initialization is beta = 0.981764955373023</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.7486043 0.9817650     1.0163779    0.17968008
std        0.1490145 0.1802539     0.2396772    0.01351221</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.3303516 0.5240838     0.3207594    0.13426792
std        0.1113180 0.1192133     0.1197515    0.03516804</code></pre>
<pre><code>          calibrated      raw internal.gwas external.gwas
point.est  0.8172142 1.910353     1.2717516    0.13610421
std        0.3731510 1.114543     0.8982115    0.03555232</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.7068027 0.8658174     0.5874275    0.14431802
std        0.3029277 0.5178111     1.0984404    0.01517989</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.706802719042964 and the other root is beta = -1.64201366979264 and the
initialization is beta = 0.706802719042964</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.706802719042964 and the other root is beta = -1.64201366982129 and the
initialization is beta = 0.706802719042964</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.706802719042964 and the other root is beta = -1.6420136698052 and the
initialization is beta = 0.706802719042964</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.865817355857281 and the other root is beta =
-1.3543083316207 and the initialization is beta = 0.865817355857281</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var),
sqrt(dat2$int.var)): The estimating equations might have another finite root.
The closest root is beta = 0.587427522237813 and the other root is beta =
-1.88125359762957 and the initialization is beta = 0.587427522237813</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var),
sqrt(dat2$int.var)): The estimating equations might have another finite root.
The closest root is beta = 0.587427522237813 and the other root is beta =
-1.8812535941219 and the initialization is beta = 0.587427522237813</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var),
sqrt(dat2$int.var)): The estimating equations might have another finite root.
The closest root is beta = 0.587427522237813 and the other root is beta =
-1.88125359591494 and the initialization is beta = 0.587427522237813</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.7068027 0.8658174     0.5874275     0.1443180
std        0.2498956 0.2359132     0.2375295     0.0165201</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-37-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="sibling-bmi-on-dbp" class="section level3">
<h3>10. Sibling: BMI on dbp</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;dbp&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(2,4))
pvals = c(1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # debiased IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_divw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_divw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_divw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_divw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;Debiased IVW: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.mle: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main= paste0(&#39;mr.raps.shrinkage: &#39;, exp, &quot; on &quot;, out, &quot;: pval=&quot;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.18003324 0.23887796    0.24103331    0.16282333
std       0.04915854 0.05338508    0.05045132    0.02328566</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.27550876 0.39087703    0.35169467     0.1671759
std       0.07831746 0.09431868    0.07549524     0.0239826</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.26209439 0.36885509    0.38629315    0.17968008
std       0.07909535 0.09333101    0.06466026    0.01218798</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.26209439 0.36885509    0.38629314    0.17968008
std       0.07289252 0.08518384    0.07373991    0.01351221</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.20471853 0.24891978    0.26786915    0.13426792
std       0.07386723 0.08019442    0.06977965    0.03516804</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est 0.25786589 0.3326803    0.33264144    0.13610421
std       0.09322744 0.1083586    0.08731208    0.03555232</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est 0.26931161 0.3373349    0.34823532    0.14431802
std       0.08371888 0.1032257    0.07739825    0.01517989</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est 0.26931161 0.3373348    0.34823532     0.1443180
std       0.08929067 0.1054270    0.08250039     0.0165201</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-41-1.png" width="1440" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin20.6.0 (64-bit)
Running under: macOS Monterey 12.0.1

Matrix products: default
BLAS:   /usr/local/Cellar/openblas/0.3.18/lib/libopenblasp-r0.3.18.dylib
LAPACK: /usr/local/Cellar/r/4.1.1_1/lib/R/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] mr.raps_0.4.1                MendelianRandomization_0.7.0
[3] workflowr_1.6.2             

loaded via a namespace (and not attached):
 [1] httr_1.4.6         sass_0.4.4         tidyr_1.2.1        jsonlite_1.7.2    
 [5] viridisLite_0.4.0  splines_4.1.1      foreach_1.5.2      bslib_0.4.1       
 [9] highr_0.9          arrangements_1.1.9 yaml_2.2.1         robustbase_0.95-1 
[13] ggrepel_0.9.3      pillar_1.9.0       lattice_0.20-44    quantreg_5.94     
[17] glue_1.4.2         digest_0.6.28      promises_1.2.0.1   iterpc_0.4.2      
[21] colorspace_2.0-2   plyr_1.8.6         htmltools_0.5.5    httpuv_1.6.3      
[25] Matrix_1.5-3       pkgconfig_2.0.3    httpcode_0.3.0     SparseM_1.81      
[29] purrr_0.3.4        scales_1.2.1       rsnps_0.5.0.0      whisker_0.4       
[33] rootSolve_1.8.2.3  later_1.3.0        MatrixModels_0.5-1 git2r_0.28.0      
[37] tibble_3.1.5       gmp_0.7-1          generics_0.1.2     ggplot2_3.4.3     
[41] ellipsis_0.3.2     cachem_1.0.6       lazyeval_0.2.2     cli_3.6.1         
[45] survival_3.2-11    magrittr_2.0.1     evaluate_0.14      fs_1.5.0          
[49] fansi_0.5.0        MASS_7.3-54        tools_4.1.1        data.table_1.14.9 
[53] lifecycle_1.0.3    stringr_1.4.0      plotly_4.10.1      munsell_0.5.0     
[57] glmnet_4.1-4       compiler_4.1.1     jquerylib_0.1.4    rlang_1.1.1       
[61] grid_4.1.1         iterators_1.0.14   rstudioapi_0.13    rjson_0.2.21      
[65] htmlwidgets_1.6.2  rmarkdown_2.11     gtable_0.3.0       codetools_0.2-18  
[69] curl_4.3.2         R6_2.5.1           gridExtra_2.3      knitr_1.36        
[73] dplyr_1.0.7        fastmap_1.1.0      utf8_1.2.2         nortest_1.0-4     
[77] rprojroot_2.0.2    shape_1.4.6        stringi_1.7.5      crul_1.4.0        
[81] Rcpp_1.0.8.3       vctrs_0.6.3        DEoptimR_1.0-13    tidyselect_1.2.0  
[85] xfun_0.27         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
