<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yunqi Yang" />


<title>UKB mr result: BMI as exposure</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">simulation-calibrated-mr</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/yunqiyang0215/simulation-calibrated-mr">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">UKB mr result: BMI as exposure</h1>
<h4 class="author">Yunqi Yang</h4>
<h4 class="date">8/6/2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-08-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>simulation-calibrated-mr/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230217code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20230217)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230217code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230217)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_
</td>
<td style="text-align:left;">
data/snp_list/bmi_
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215simulationcalibratedmrtreed1323c0bb3b797bfd051b597066aeb7ab0b65af1targetblankd1323c0a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/tree/d1323c0bb3b797bfd051b597066aeb7ab0b65af1" target="_blank">d1323c0</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215simulationcalibratedmrtreed1323c0bb3b797bfd051b597066aeb7ab0b65af1targetblankd1323c0a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/tree/d1323c0bb3b797bfd051b597066aeb7ab0b65af1" target="_blank">d1323c0</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    scripts/.DS_Store

Untracked files:
    Untracked:  output/ukb_mr_sib1.pdf
    Untracked:  output/ukb_mr_sib2.pdf
    Untracked:  output/ukb_mr_trio1.pdf

Unstaged changes:
    Deleted:    scripts/real_data_mr_bmi.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/ukb_mr_bmi_202407.Rmd</code>) and HTML (<code>docs/ukb_mr_bmi_202407.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/d1323c0bb3b797bfd051b597066aeb7ab0b65af1/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">d1323c0</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-13
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/2919327adb86772679cdfb5019f162dfaae45916/docs/ukb_mr_bmi_202407.html" target="_blank">2919327</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/40643d5cdf22f2e10612f64d5a0ed90c895dd147/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">40643d5</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/ceb7aae41766ff85e99caf85621cbf766ab389a0/docs/ukb_mr_bmi_202407.html" target="_blank">ceb7aae</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/c2285a525f132517309bbc2c2f14a85ef170728e/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">c2285a5</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/47e777a9d3974dcd94c63cfeac246bca3225ba56/docs/ukb_mr_bmi_202407.html" target="_blank">47e777a</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/baa03061f7734cc1f1c811351edbac75670aecc4/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">baa0306</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/711b69207310a97738a72fe4d53a67b9e02cae57/docs/ukb_mr_bmi_202407.html" target="_blank">711b692</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/2ab18942089b73b65482166b0826fff8fd1a7ccb/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">2ab1894</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/818190488ce7da0f341fa7b5807a7f8817a01c6f/docs/ukb_mr_bmi_202407.html" target="_blank">8181904</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/981aeb1eea621c7190c904be5a4e9e0eb5d1610a/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">981aeb1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/d17ce9773a200d70f36a81c6e4f8158ba5603455/docs/ukb_mr_bmi_202407.html" target="_blank">d17ce97</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/ce497d3f699eb69d0a7ecd3862567282d298b7ef/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">ce497d3</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/b26f07491f05dc3cdf48e643fa4b7c69966fcfce/docs/ukb_mr_bmi_202407.html" target="_blank">b26f074</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/e2b5caca6796af8957601a6289a31eace7d69c44/docs/ukb_mr_bmi_202407.html" target="_blank">e2b5cac</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/e3f8ab9d4abdf67232857b4311bfd9e2b4a3f9a1/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">e3f8ab9</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/2b0c94c3a51580c5ce73113863de7dcdd685fbc7/docs/ukb_mr_bmi_202407.html" target="_blank">2b0c94c</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/c2819fa100af91266389c8e5cd71a7072e838c86/docs/ukb_mr_bmi_202407.html" target="_blank">c2819fa</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/f53447f0e212bf11f19c1f9bf689ca6c89e5a724/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">f53447f</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/fe15e0731151212f52ec1d0c9b3c09a7cd8c589c/docs/ukb_mr_bmi_202407.html" target="_blank">fe15e07</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/0dc3b0bae18b0269c113ecb92ad4aad75d91b55b/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">0dc3b0b</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/simulation-calibrated-mr/1584f2809565e293474f8d371a271cb50a587041/docs/ukb_mr_bmi_202407.html" target="_blank">1584f28</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/8958ff17ef2ae2da9a9a0889d2d4b41c6f3f2c94/analysis/ukb_mr_bmi_202407.Rmd" target="_blank">8958ff1</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
<td>
wflow_publish("analysis/ukb_mr_bmi_202407.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="snp-selection-file-available-at" class="section level4">
<h4>SNP selection file available at:</h4>
<p><a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/master/analysis/snp_selection.R" class="uri">https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/master/analysis/snp_selection.R</a></p>
<pre class="r"><code>library(MendelianRandomization)
library(mr.raps)</code></pre>
</div>
<div id="trio-bmi-on-diabetes" class="section level3">
<h3>1. Trio: BMI on diabetes</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;diabetes&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(3,3))
pvals = c(1e-2, 1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: BMI on diabetes: pval=&#39;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.mle: BMI on diabetes: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.shrinkage: BMI on diabetes: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est  0.1742372 0.08310121    0.09718587     0.2145300
std        0.0348829 0.03411473    0.03607534     0.0155471</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.8526724 1.1381238     0.8332563   0.261556673
std        0.2874202 0.5623971     0.9123381   0.009846032</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.852672462956953 and the other root is beta = -1.39667137923214 and the
initialization is beta = 0.852672448208213</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.852672462956953 and the other root is beta = -1.39667137965123 and the
initialization is beta = 0.852672448208213</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 1.13812384334091 and the other root is beta =
-1.05476338394564 and the initialization is beta = 1.13812379978625</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var),
sqrt(dat2$int.var)): The estimating equations might have another finite root.
The closest root is beta = 0.833256304338921 and the other root is beta =
-1.20904015180807 and the initialization is beta = 0.83325627221157</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.8526725 1.1381238     0.8332563    0.26155667
std        0.2179483 0.3883543     0.3210029    0.01173057</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.19199294 0.14767053    0.16415084     0.2136415
std       0.07049309 0.07261298    0.07206222     0.0295015</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.6377509 0.9228341     0.6800884    0.25244038
std        0.2189662 0.5609100     1.8210034    0.01181858</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.637750917841163 and the other root is beta = -3.24017916957212 and the
initialization is beta = 0.637750935727596</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.637750917841163 and the other root is beta = -3.24017913449575 and the
initialization is beta = 0.637750935727596</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.637750917841163 and the other root is beta = -3.24017907353752 and the
initialization is beta = 0.637750935727596</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var),
sqrt(dat2$int.var)): The estimating equations might have another finite root.
The closest root is beta = 0.680088420908623 and the other root is beta =
-1.4863958670254 and the initialization is beta = 0.680088420908623</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var),
sqrt(dat2$int.var)): The estimating equations might have another finite root.
The closest root is beta = 0.680088420908623 and the other root is beta =
-1.48639606681452 and the initialization is beta = 0.680088420908623</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.6377509 0.9228341     0.6800884    0.25244038
std        0.2982836 0.4323314     0.3151571    0.01432002</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.2952433 0.1375211     0.1432149    0.21792086
std        0.1138949 0.1110828     0.1144336    0.05425065</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.6510441 0.5372329      0.325047    0.26155270
std        0.2402355 0.3406265      1.013362    0.01463554</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.537232941299394 and the other root is beta =
-2.22445573374387 and the initialization is beta = 0.537232941299394</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.537232941299394 and the other root is beta =
-2.22445928418777 and the initialization is beta = 0.537232941299394</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.6510441 0.5372329     0.3250470    0.26155270
std        0.3190874 0.4287565     0.2658774    0.01838022</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-5-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/47e777a9d3974dcd94c63cfeac246bca3225ba56/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-5-1.png" target="_blank">47e777a</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/711b69207310a97738a72fe4d53a67b9e02cae57/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-5-1.png" target="_blank">711b692</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/d17ce9773a200d70f36a81c6e4f8158ba5603455/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-5-1.png" target="_blank">d17ce97</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>par(mfrow = c(1,3))
pvals = c(1e-2, 1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # Debiased IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_divw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_divw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_divw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_divw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;Debiased IVW: BMI on diabetes: pval=&#39;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
}</code></pre>
<pre><code>          calibrated      raw internal.gwas external.gwas
point.est  1.3827611 1.380499      1.967138    0.23040006
std        0.7602672 2.026258      2.125639    0.01708713</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.7034634 0.8917999      1.794472    0.21935251
std        0.3580443 0.6988244      2.104226    0.03041011</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.7451782 0.4178795      0.563110    0.22090121
std        0.4135047 0.3845947      0.560563    0.05489029</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-6-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="sibling-bmi-on-diabetes" class="section level3">
<h3>2. Sibling: BMI on diabetes</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;diabetes&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(3,3))
pvals = c(1e-2, 1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: BMI on diabetes: pval=&#39;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.mle: BMI on diabetes: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.shrinkage: BMI on diabetes: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.18096589 0.14170779    0.11912744     0.2145300
std       0.03296944 0.03389564    0.02865946     0.0155471</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.50107091 0.40413012    0.28804194   0.261556673
std       0.07693795 0.09451106    0.05336567   0.009846032</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), :
The estimating equations might have another finite root. The closest root is
beta = 0.501070925798636 and the other root is beta = -2.329156228529 and the
initialization is beta = 0.501070913826496</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.501070925798636 and the other root is beta = -2.32915612824109 and the
initialization is beta = 0.501070913826496</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.501070925798636 and the other root is beta = -2.32915626196206 and the
initialization is beta = 0.501070913826496</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.50107093 0.40413011    0.28804194    0.26155667
std       0.08830563 0.09587784    0.06568466    0.01173057</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.21049994 0.18066066    0.18289797     0.2136415
std       0.05647031 0.05935615    0.04844138     0.0295015</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.38186359 0.34226803    0.29933464    0.25244038
std       0.07083702 0.08267482    0.05999259    0.01181858</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.381863582680002 and the other root is beta = -3.03773169939349 and the
initialization is beta = 0.38186359316658</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est 0.38186358 0.3422680    0.29933464    0.25244038
std       0.08976942 0.1016511    0.07103329    0.01432002</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.33130600 0.34944851    0.22314574    0.21792086
std       0.08999924 0.09149393    0.08083187    0.05425065</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est 0.50412307 0.5189957    0.33804200    0.26155270
std       0.08335365 0.1035925    0.06983543    0.01463554</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.504123068547964 and the other root is beta = -2.33412796877415 and the
initialization is beta = 0.504123073461214</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.518995682071063 and the other root is beta =
-2.40749590986373 and the initialization is beta = 0.518995682071063</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.5041231 0.5189957    0.33804200    0.26155270
std        0.1057312 0.1181563    0.09166537    0.01838022</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-10-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow = c(1,3))
pvals = c(1e-2, 1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # Debiased IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_divw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_divw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_divw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_divw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;Debiased IVW: BMI on diabetes: pval=&#39;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
}</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est 0.45362967 0.3984430    0.25204762    0.23040006
std       0.09004477 0.1024335    0.06250326    0.01708713</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.32237072 0.29611540    0.26670543    0.21935251
std       0.08823803 0.09920197    0.07178127    0.03041011</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est   0.418180 0.4683126     0.2770039    0.22090121
std         0.114872 0.1255145     0.1006931    0.05489029</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-11-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/d17ce9773a200d70f36a81c6e4f8158ba5603455/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-11-1.png" target="_blank">d17ce97</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/e2b5caca6796af8957601a6289a31eace7d69c44/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-11-1.png" target="_blank">e2b5cac</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/c2819fa100af91266389c8e5cd71a7072e838c86/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-11-1.png" target="_blank">c2819fa</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/fe15e0731151212f52ec1d0c9b3c09a7cd8c589c/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-11-1.png" target="_blank">fe15e07</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/1584f2809565e293474f8d371a271cb50a587041/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-11-1.png" target="_blank">1584f28</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="trio-bmi-on-education-years" class="section level3">
<h3>3. Trio: BMI on education years</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;education_yrs&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(3,3))
pvals = c(1e-2, 1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: BMI on EduYrs: pval=&#39;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.mle: BMI on EduYrs: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.shrinkage: BMI on EduYrs: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.02862617 0.08263537    0.04640938   -0.14096929
std       0.05751157 0.06078472    0.05960953    0.01492236</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.5573831 2.0077055     0.6308491    -0.1627887
std        0.3196189 0.8887191     2.2539519     0.0111244</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 2.00770554992002 and the other root is beta =
-0.941963127811282 and the initialization is beta = 2.00770554992002</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 2.00770554992002 and the other root is beta =
-0.941963176969064 and the initialization is beta = 2.00770554992002</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var),
sqrt(dat2$int.var)): The estimating equations might have another finite root.
The closest root is beta = 0.630849095237435 and the other root is beta =
-5.24081360585966 and the initialization is beta = 0.630849095237435</code></pre>
<pre><code>          calibrated      raw internal.gwas external.gwas
point.est  0.5573831 2.007706     0.6308491   -0.16278875
std        1.2108488 1.516290     0.8121806    0.01236329</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.2298938 0.1893570   -0.01709613   -0.12169550
std        0.1014469 0.1129532    0.11970946    0.02253355</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.5413199 0.7430371     -0.150400   -0.13133694
std        0.5093184 1.2533918      2.102216    0.01351656</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= 0.541319940126336 and the other root is beta = -4.86989246701021 and the
initialization is beta = 0.541319940126336</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.743037052466926 and the other root is beta =
-4.13456502687486 and the initialization is beta = 0.743037052466926</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.5413199 0.7430371    -0.1503999    -0.1313369
std        0.2447361 0.4627413     1.0553844     0.0145233</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.1222599 0.1339975    0.05738363   -0.08705489
std        0.1627665 0.1810404    0.19745066    0.03096505</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.2104791 0.2852621     0.2960719   -0.09119262
std        0.6323016 1.6475888     0.6967661    0.01690865</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.2104791 0.2852621     0.2960719   -0.09119262
std        0.2815900 0.3890766     0.9749314    0.01772449</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-15-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="sibling-bmi-on-education-years" class="section level3">
<h3>4. Sibling: BMI on education years</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;education_yrs&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(3,3))
pvals = c(1e-2, 1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: BMI on EduYrs: pval=&#39;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.mle: BMI on EduYrs: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.shrinkage: BMI on EduYrs: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.01819767 -0.03808504   -0.18002453   -0.14096929
std        0.03537125  0.03647938    0.03459958    0.01492236</code></pre>
<pre><code>           calibrated        raw internal.gwas external.gwas
point.est -0.04373665 -0.1012368    -0.3896846    -0.1627887
std        0.09195833  0.1077901     0.0715704     0.0111244</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.04373665 -0.10123678   -0.38968455   -0.16278875
std        0.08502516  0.09703345    0.07414422    0.01236329</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.01825871 -0.03786022   -0.20781912   -0.12169550
std        0.05513444  0.05826400    0.05312938    0.02253355</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.02591777 -0.05723418    -0.2969445   -0.13133694
std        0.08913997  0.10134807     0.0800222    0.01351656</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.02591777 -0.05723418   -0.29694452    -0.1313369
std        0.07827802  0.08814827    0.07741541     0.0145233</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.02791726 -0.05763980   -0.12027563   -0.08705489
std        0.07464877  0.08238804    0.07331826    0.03096505</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.03388451 -0.07216306   -0.14327112   -0.09119262
std        0.09620649  0.11514805    0.09396632    0.01690865</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.03388451 -0.07216306   -0.14327112   -0.09119262
std        0.09064139  0.10329808    0.08783489    0.01772449</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-19-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="trio-bmi-on-smoking" class="section level3">
<h3>5. Trio: BMI on smoking</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;smoking&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(3,3))
pvals = c(1e-2, 1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: BMI on smoking: pval=&#39;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.mle: BMI on smoking: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.shrinkage: BMI on smoking: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>           calibrated         raw internal.gwas external.gwas
point.est -0.06965083 -0.01090133   -0.03198029    0.10817601
std        0.03532488  0.03657556    0.03767364    0.01186287</code></pre>
<pre><code>Warning in sqrt(1/I): NaNs produced</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est -0.5119636 -0.1834146    -0.2845334    0.12104782
std        0.2714343  0.5096724     8.8139215    0.01005045</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= -0.511963589483925 and the other root is beta = 3.21852996882967 and the
initialization is beta = -0.51196361452806</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= -0.511963589483925 and the other root is beta = 3.21852758607933 and the
initialization is beta = -0.51196361452806</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), : The
estimating equations might have another finite root. The closest root is beta
= -0.511963589483925 and the other root is beta = 3.21853004436683 and the
initialization is beta = -0.51196361452806</code></pre>
<pre><code>Warning in sqrt(1/I): NaNs produced

Warning in sqrt(1/I): NaNs produced

Warning in sqrt(1/I): NaNs produced</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est -0.5119636 -0.1834146    -0.2845334    0.12104782
std        0.2999038  0.6214294     0.3357300    0.01068275</code></pre>
<pre><code>           calibrated        raw internal.gwas external.gwas
point.est -0.07064455 0.07046356   -0.00335690    0.09231808
std        0.06517653 0.07512403    0.07547593    0.01650998</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est -0.2603259 0.6172662   -0.03188522    0.09708124
std        0.1940594 0.4531537    0.94481116    0.01219272</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.617266212779414 and the other root is beta =
-2.03489584409894 and the initialization is beta = 0.617266212779414</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.617266212779414 and the other root is beta =
-2.03489591583129 and the initialization is beta = 0.617266212779414</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.617266212779414 and the other root is beta =
-2.0348959370985 and the initialization is beta = 0.617266212779414</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est -0.2603259 0.6172662   -0.03188524    0.09708124
std        0.2623595 0.6230411    0.71674875    0.01263152</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est 0.08032099 0.1656224   -0.06955929    0.06982057
std       0.10853475 0.1205187    0.11984137    0.02554330</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.2027309 0.5031254    -0.2219370    0.07256776
std        0.2372671 0.4569862     0.5907086    0.01514555</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.503125377566961 and the other root is beta =
-2.61255850437002 and the initialization is beta = 0.503125377566961</code></pre>
<pre><code>Warning in mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var),
sqrt(dat2$raw.var)): The estimating equations might have another finite root.
The closest root is beta = 0.503125377566961 and the other root is beta =
-2.61255850543813 and the initialization is beta = 0.503125377566961</code></pre>
<pre><code>          calibrated       raw internal.gwas external.gwas
point.est  0.2027309 0.5031254    -0.2219370    0.07256776
std        0.2671649 0.3825085     0.3847335    0.01571885</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-23-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="sibling-bmi-on-smoking" class="section level3">
<h3>6. Sibling: BMI on smoking</h3>
<pre class="r"><code>exp = &quot;bmi&quot;
out = &quot;smoking&quot;</code></pre>
<pre class="r"><code># exposure trait
dat.int1 &lt;- list()
dat.ext1 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, exp, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, exp, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 &lt;- do.call(rbind, dat.int1)
dat.ext1 &lt;- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)</code></pre>
<pre class="r"><code># outcome trait
dat.int2 &lt;- list()
dat.ext2 &lt;- list()

for (chr_num in 1:22) {
  f1 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/&quot;, out, &quot;_chr_&quot;, chr_num, &quot;.txt&quot;)
  f2 &lt;- paste0(&quot;/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr&quot;, chr_num, &quot;.&quot;, out, &quot;.glm.linear&quot;)
  
  int &lt;- read.csv(f1, header = TRUE, sep = &quot; &quot;)
  ext &lt;- read.csv(f2, header = TRUE, sep = &quot;\t&quot;)
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 &lt;- do.call(rbind, dat.int2)
dat.ext2 &lt;- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = &quot;variant&quot;, by.y = &quot;ID&quot;, all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)</code></pre>
<pre class="r"><code>par(mfrow = c(3,3))
pvals = c(1e-2, 1e-5, 1e-8)

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0(&quot;/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/bmi_&quot;,
                                          pvals[i], &quot;.txt&quot;), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # IVW
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  MRInput &lt;- mr_input(bx = dat1$cali,
                      bxse = sqrt(dat1$cali.var),
                      by = dat2$cali,
                      byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[, 1] = c(mr@Estimate, mr@StdError)
  
  
  # uncalibrated
  MRInput &lt;- mr_input(bx = dat1$raw,
                      bxse = sqrt(dat1$raw.var),
                      by = dat2$raw,
                      byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[, 2] = c(mr@Estimate, mr@StdError)
  
  
  # internal gwas
  MRInput &lt;- mr_input(bx = dat1$int,
                      bxse = sqrt(dat1$int.var),
                      by = dat2$int,
                      byse = sqrt(dat2$int.var))
  mr = mr_ivw(MRInput)
  res.mr[, 3] = c(mr@Estimate, mr@StdError)
  
  
  # external gwas
  MRInput &lt;- mr_input(bx = dat1$BETA,
                      bxse = dat1$SE,
                      by = dat2$BETA,
                      byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[, 4] = c(mr@Estimate, mr@StdError)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;IVW: BMI on smoking: pval=&#39;, pvals[i]) , cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  
  # mr.raps.mle
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.mle(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.mle: BMI on smoking: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
  
  ## mr.raps.shrinkage
  res.mr = matrix(NA, ncol = 4, nrow = 2)
  colnames(res.mr) = c(&quot;calibrated&quot;, &quot;raw&quot;, &quot;internal.gwas&quot;, &quot;external.gwas&quot;)
  rownames(res.mr) = c(&quot;point.est&quot;, &quot;std&quot;)
  
  # calibrated
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 1] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 2] = c(point_est, stderr)
  
  # internal gwas
  mr = mr.raps.shrinkage(dat1$int, dat2$int, sqrt(dat1$int.var), sqrt(dat2$int.var))
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 3] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[, 4] = c(point_est, stderr)
  print(res.mr)
  
  labels &lt;- c(&#39;calibrated&#39;, &#39;raw&#39;, &#39;internal.gwas&#39;, &#39;external.gwas&#39;)
  point_est &lt;- res.mr[1, ]
  error_bars &lt;- 1.96 * res.mr[2, ]
  
  # Plotting
  plot(1:4, point_est, ylim=c(min(point_est - error_bars), max(point_est + error_bars)),
       xaxt=&#39;n&#39;, pch=19, xlab=&#39;&#39;, ylab=&#39;Point Estimate&#39;, main=paste0(&#39;mr.raps.shrinkage: BMI on smoking: pval=&#39;, pvals[i]), cex=1.5)
  arrows(1:4, point_est - error_bars, 1:4, point_est + error_bars, angle=90, code=3, length=0.1, col=&#39;red&#39;)
  axis(1, at=1:4, labels=labels)
    
}</code></pre>
<pre><code>          calibrated         raw internal.gwas external.gwas
point.est 0.04494429 0.004231541    0.07320474    0.10817601
std       0.03242624 0.033757422    0.02731122    0.01186287</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.12406843 0.01152000    0.14939321    0.12104782
std       0.07414012 0.09719242    0.05984378    0.01005045</code></pre>
<pre><code>          calibrated        raw internal.gwas external.gwas
point.est 0.12406844 0.01151998    0.14939321    0.12104782
std       0.08705832 0.09189509    0.05603064    0.01068275</code></pre>
<pre><code>          calibrated         raw internal.gwas external.gwas
point.est 0.01627286 -0.02166770    0.10226795    0.09231808
std       0.05156174  0.05475607    0.04762585    0.01650998</code></pre>
<pre><code>          calibrated         raw internal.gwas external.gwas
point.est 0.02621466 -0.03598158    0.16310431    0.09708124
std       0.07263472  0.08696831    0.06087014    0.01219272</code></pre>
<pre><code>          calibrated         raw internal.gwas external.gwas
point.est 0.02621466 -0.03598157    0.16310431    0.09708124
std       0.07903666  0.08950178    0.06971466    0.01263152</code></pre>
<pre><code>          calibrated          raw internal.gwas external.gwas
point.est 0.03078754 -0.005483624    0.02498641    0.06982057
std       0.07848377  0.095001021    0.06766562    0.02554330</code></pre>
<pre><code>          calibrated          raw internal.gwas external.gwas
point.est 0.04271851 -0.008889594    0.03305786    0.07256776
std       0.07925882  0.091617541    0.07214723    0.01514555</code></pre>
<pre><code>          calibrated          raw internal.gwas external.gwas
point.est 0.04271851 -0.008889599    0.03305786    0.07256776
std       0.09202153  0.123183937    0.07939483    0.01571885</code></pre>
<p><img src="figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-27-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/d17ce9773a200d70f36a81c6e4f8158ba5603455/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-27-1.png" target="_blank">d17ce97</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/fe15e0731151212f52ec1d0c9b3c09a7cd8c589c/docs/figure/ukb_mr_bmi_202407.Rmd/unnamed-chunk-27-1.png" target="_blank">fe15e07</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2024-08-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin20.6.0 (64-bit)
Running under: macOS Monterey 12.0.1

Matrix products: default
BLAS:   /usr/local/Cellar/openblas/0.3.18/lib/libopenblasp-r0.3.18.dylib
LAPACK: /usr/local/Cellar/r/4.1.1_1/lib/R/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] mr.raps_0.4.1                MendelianRandomization_0.7.0
[3] workflowr_1.6.2             

loaded via a namespace (and not attached):
 [1] httr_1.4.6         sass_0.4.4         tidyr_1.2.1        jsonlite_1.7.2    
 [5] viridisLite_0.4.0  splines_4.1.1      foreach_1.5.2      bslib_0.4.1       
 [9] highr_0.9          arrangements_1.1.9 yaml_2.2.1         robustbase_0.95-1 
[13] ggrepel_0.9.3      pillar_1.9.0       lattice_0.20-44    quantreg_5.94     
[17] glue_1.4.2         digest_0.6.28      promises_1.2.0.1   iterpc_0.4.2      
[21] colorspace_2.0-2   plyr_1.8.6         htmltools_0.5.5    httpuv_1.6.3      
[25] Matrix_1.5-3       pkgconfig_2.0.3    httpcode_0.3.0     SparseM_1.81      
[29] purrr_0.3.4        scales_1.2.1       rsnps_0.5.0.0      whisker_0.4       
[33] rootSolve_1.8.2.3  later_1.3.0        MatrixModels_0.5-1 git2r_0.28.0      
[37] tibble_3.1.5       gmp_0.7-1          generics_0.1.2     ggplot2_3.4.3     
[41] ellipsis_0.3.2     cachem_1.0.6       lazyeval_0.2.2     cli_3.6.1         
[45] survival_3.2-11    magrittr_2.0.1     evaluate_0.14      fs_1.5.0          
[49] fansi_0.5.0        MASS_7.3-54        tools_4.1.1        data.table_1.14.9 
[53] lifecycle_1.0.3    stringr_1.4.0      plotly_4.10.1      munsell_0.5.0     
[57] glmnet_4.1-4       compiler_4.1.1     jquerylib_0.1.4    rlang_1.1.1       
[61] grid_4.1.1         iterators_1.0.14   rstudioapi_0.13    rjson_0.2.21      
[65] htmlwidgets_1.6.2  rmarkdown_2.11     gtable_0.3.0       codetools_0.2-18  
[69] curl_4.3.2         R6_2.5.1           gridExtra_2.3      knitr_1.36        
[73] dplyr_1.0.7        fastmap_1.1.0      utf8_1.2.2         nortest_1.0-4     
[77] rprojroot_2.0.2    shape_1.4.6        stringi_1.7.5      crul_1.4.0        
[81] Rcpp_1.0.8.3       vctrs_0.6.3        DEoptimR_1.0-13    tidyselect_1.2.0  
[85] xfun_0.27         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
