---
title: "Perform MR to assess causal effect estimate"
author: "Yunqi Yang"
date: "5/15/2023"
output: html_document
---

### Description: 

Perform MR to assess causal effect estimate. We should focus on both effect size estimate and standard error of the causal effect. Here the sample size for external data $N=1e4$ and internal data $n=1e3$. 

```{r}
library(MendelianRandomization)
dat = readRDS("./data/realdata.rds")
source("./code/calibrated_mr.R")
```

### Compute raw and calibrated estimator for each SNP, adjusted for non-transmitted allele
```{r}
n = 1e3
p = 1e3
N = 1e4

y1.cali <- matrix(NA, nrow = p, ncol = 2)
y1.raw <- matrix(NA, nrow = p, ncol = 2)
colnames(y1.cali) <- c("point_estimate", "var_estimate")
colnames(y1.raw) <- c("point_estimate", "var_estimate")

y2.cali <- matrix(NA, nrow = p, ncol = 2)
y2.raw <- matrix(NA, nrow = p, ncol = 2)
colnames(y2.cali) <- c("point_estimate", "var_estimate")
colnames(y2.raw) <- c("point_estimate", "var_estimate")
```


```{r }
for (i in 1:p){
  dat1_valid <- cbind(dat[["pheno"]][1:n,1], dat[["trans"]][1:n, i], dat[["nontrans"]][1:n, i]) 
  dat1_ext <- cbind(dat[["pheno"]][,1], dat[["trans"]][, i]) 
  
  coef_ext <- compute_sumstat(dat1_ext[,1], dat1_ext[,2])$bhat
  res_int <- compute_sumstat(dat1_valid[,1], dat1_valid[,2])
  res_int_adj <- compute_adj_sumstat(dat1_valid[,1], dat1_valid[,2], dat1_valid[,3])
  res1 <- calibrated_estimator(dat1_valid, N, res_int$resid, res_int_adj$resid, res_int$bhat, res_int_adj$bhat, coef_ext)

  
  # Select different samples for outcome trait. 
  dat2_valid <- cbind(dat[["pheno"]][(1+1e3):(n+1e3),2], dat[["trans"]][(1+1e3):(n+1e3), i], dat[["nontrans"]][(1+1e3):(n+1e3), i]) 
  dat2_ext <- cbind(dat[["pheno"]][, 2], dat[["trans"]][, i]) 
  
  coef_ext <- compute_sumstat(dat2_ext[,1], dat2_ext[,2])$bhat
  res_int <- compute_sumstat(dat2_valid[,1], dat2_valid[,2])
  res_int_adj <- compute_adj_sumstat(dat2_valid[,1], dat2_valid[,2], dat2_valid[,3])
  res2 <- calibrated_estimator(dat2_valid, N, res_int$resid, res_int_adj$resid, res_int$bhat, res_int_adj$bhat, coef_ext)
  
  
  y1.cali[i, ] = c(res1$tau_cal, res1$var_tau_cal)
  y2.cali[i, ] = c(res2$tau_cal, res2$var_tau_cal)
  y1.raw[i, ] = c(res1$tau_raw, res1$var_tau_raw)
  y2.raw[i, ] = c(res2$tau_raw, res2$var_tau_raw)
}

head(y1.cali)
head(y1.raw)
head(y2.cali)
head(y2.raw)
```

### Compute unadjusted SNP effect size & standard errors 

This is just to calculate GWAS summary stat from external data. 
```{r}
n = 1e3
p = 1e3
N = 1e4

y1 <- matrix(NA, nrow = p, ncol = 2)
y2 <- matrix(NA, nrow = p, ncol = 2)

colnames(y1) <- c("point_estimate", "std.error")
colnames(y2) <- c("point_estimate", "std.error")

for (i in 1:p){
  dat1_ext <- cbind(dat[["pheno"]][,1], dat[["trans"]][, i]) 
  dat2_ext <- cbind(dat[["pheno"]][, 2], dat[["trans"]][, i]) 
  fit1 <- lm(dat1_ext[,1] ~ dat1_ext[,2])
  fit2 <- lm(dat2_ext[,1] ~ dat2_ext[,2])
  
  y1[i, ] = c(summary(fit1)$coef[2,1], summary(fit1)$coef[2,2])
  y2[i, ] = c(summary(fit2)$coef[2,1], summary(fit2)$coef[2,2])
}
```

```{r }
par(mfrow = c(1,2))
plot(y1.cali[,2], y1.raw[,2], pch = 20, xlab = "calibrated estimator", ylab = "raw estimator",  main = "SNP variance: exposure trait")
abline(a = 0, b = 1, col = "red")

plot(y2.cali[,2], y2.raw[,2], pch = 20, xlab = "calibrated estimator", ylab = "raw estimator", main = "SNP variance: outcome trait")
abline(a = 0, b = 1, col = "red")
```

### Calculate z-scores 
```{r}
z2p <- function(zscore){
  pval <- pnorm(zscore, mean = 0, sd = 1, lower.tail = FALSE)*2
  return(pval)
}
```

```{r}
z1.cali = y1.cali[,1]/sqrt(y1.cali[,2])
z1.raw = y1.raw[,1]/sqrt(y1.raw[,2])
z2.cali = y2.cali[,1]/sqrt(y2.cali[,2])
z2.raw = y2.raw[,1]/sqrt(y2.raw[,2])

z1 = y1[, 1]/y1[,2]
z2 = y2[, 1]/y2[,2]
  
  
p1.cali = z2p(z1.cali)
p1.raw = z2p(z1.raw)
p2.cali = z2p(z2.cali)
p2.raw = z2p(z2.raw)

p1 = z2p(z1)
p2 = z2p(z2)
```

```{r}
par(mfrow = c(2,2))
plot(-log10(p1.cali), pch = 20, cex = 0.8, ylab = "-log10(pval)", main = "calibrated exposure")
plot(-log10(p2.cali), pch = 20, cex = 0.8, ylab = "-log10(pval)", main = "calibrated outcome")
plot(-log10(p1.raw), pch = 20, cex = 0.8, ylab = "-log10(pval)", main = "raw exposure")
plot(-log10(p2.raw), pch = 20, cex = 0.8, ylab = "-log10(pval)", main = "raw outcome")
```


```{r}
par(mfrow = c(1,2))
plot(-log10(p1), pch = 20, cex = 0.8, ylab = "-log10(pval)", main = "unadjusted exposure")
plot(-log10(p2), pch = 20, cex = 0.8, ylab = "-log10(pval)", main = "unadjusted outcome")

```




### Perform different MR methods to assess the causal effect estimate.
```{r}

MRInput.cali <- mr_input(bx = y1.cali[,1],
                          bxse = sqrt(y1.cali[,2]),
                          by = y2.cali[,1],
                          byse = sqrt(y2.cali[,2]))

MRInput.raw <- mr_input(bx = y1.raw[,1],
                          bxse = sqrt(y1.raw[,2]),
                          by = y2.raw[,1],
                          byse = sqrt(y2.raw[,2]))


egger.cali <- mr_egger(MRInput.cali)
egger.raw <- mr_egger(MRInput.raw)

ivw.cali <- mr_ivw(MRInput.cali)
ivw.raw <- mr_ivw(MRInput.raw)
```

```{r}
egger.cali
egger.raw
ivw.cali
ivw.raw
```

```{r}

MRInput <- mr_input(bx = y1[,1],
                          bxse = y1[,2],
                          by = y2[,1],
                          byse = y2[,2])
egger <- mr_egger(MRInput)
ivw <- mr_ivw(MRInput)

egger
ivw
```
