---
title: "Perform MR to assess causal effect estimate"
author: "Yunqi Yang"
date: "5/15/2023"
output: html_document
---

### Description: 

Perform MR to assess causal effect estimate. We should focus on both effect size estimate and standard error of the causal effect. Here the sample size for external data $N=1e4$ and internal data $n=1e3$. 

```{r}
library(MendelianRandomization)
dat = readRDS("./data/realdata.rds")
source("./code/calibrated_mr.R")
```

### 1. Compute raw and calibrated estimator for each SNP, adjusted for non-transmitted allele
```{r}
n = 1e3
p = 100
N = 1e4

y1.cali <- matrix(NA, nrow = p, ncol = 2)
y1.raw <- matrix(NA, nrow = p, ncol = 2)
colnames(y1.cali) <- c("point_estimate", "var_estimate")
colnames(y1.raw) <- c("point_estimate", "var_estimate")

y2.cali <- matrix(NA, nrow = p, ncol = 2)
y2.raw <- matrix(NA, nrow = p, ncol = 2)
colnames(y2.cali) <- c("point_estimate", "var_estimate")
colnames(y2.raw) <- c("point_estimate", "var_estimate")
```


```{r }
for (i in 1:p){
  dat1_valid <- cbind(dat[["pheno"]][1:n,1], dat[["trans"]][1:n, i], dat[["nontrans"]][1:n, i]) 
  dat1_ext <- cbind(dat[["pheno"]][,1], dat[["trans"]][, i]) 
  
  coef_ext <- compute_sumstat(dat1_ext[,1], dat1_ext[,2])$bhat
  res_int <- compute_sumstat(dat1_valid[,1], dat1_valid[,2])
  res_int_adj <- compute_adj_sumstat(dat1_valid[,1], dat1_valid[,2], dat1_valid[,3])
  res1 <- calibrated_estimator(dat1_valid, N, res_int$resid, res_int_adj$resid, res_int$bhat, res_int_adj$bhat, coef_ext)

  
  # Select different samples for outcome trait. 
  dat2_valid <- cbind(dat[["pheno"]][(1+1e3):(n+1e3),2], dat[["trans"]][(1+1e3):(n+1e3), i], dat[["nontrans"]][(1+1e3):(n+1e3), i]) 
  dat2_ext <- cbind(dat[["pheno"]][, 2], dat[["trans"]][, i]) 
  
  coef_ext <- compute_sumstat(dat2_ext[,1], dat2_ext[,2])$bhat
  res_int <- compute_sumstat(dat2_valid[,1], dat2_valid[,2])
  res_int_adj <- compute_adj_sumstat(dat2_valid[,1], dat2_valid[,2], dat2_valid[,3])
  res2 <- calibrated_estimator(dat2_valid, N, res_int$resid, res_int_adj$resid, res_int$bhat, res_int_adj$bhat, coef_ext)
  
  
  y1.cali[i, ] = c(res1$tau_cal, res1$var_tau_cal)
  y2.cali[i, ] = c(res2$tau_cal, res2$var_tau_cal)
  y1.raw[i, ] = c(res1$tau_raw, res1$var_tau_raw)
  y2.raw[i, ] = c(res2$tau_raw, res2$var_tau_raw)
}

head(y1.cali)
head(y1.raw)
head(y2.cali)
head(y2.raw)
```

```{r }
par(mfrow = c(1,2))
plot(y1.cali[,2], y1.raw[,2], pch = 20, xlab = "calibrated estimator", ylab = "raw estimator",  main = "SNP variance: exposure trait")
abline(a = 0, b = 1, col = "red")

plot(y2.cali[,2], y2.raw[,2], pch = 20, xlab = "calibrated estimator", ylab = "raw estimator", main = "SNP variance: outcome trait")
abline(a = 0, b = 1, col = "red")
```


### 2. Check variance reduction is similar to previous simulation results on single SNP

See https://yunqiyang0215.github.io/simulation-calibrated-mr/empirical_var_reduction.html

The variance reduction formula is:
$$
VR = \frac{(1-r)(1-\rho)}{2}\frac{\sigma^2}{\sigma^2+(1-r^2)\beta_2^2}.
$$


In this case, $T$ and $NT$ are independent, so $r=0$. $\sigma^2=0.01, \rho=0.1$. $\beta_2=0.001$. 

This is a rough sanity check. The variance reduction here agrees with the theoretical value, even though SNPs are not iid. 
```{r}
rho = 0.1
sigma2 = 0.1
b2 = 0.001
v = (1-rho)/2*sigma2/(sigma2+b2^2)
par(mfrow = c(1,2))
vr1 = (y1.raw[,2] - y1.cali[,2])/y1.raw[,2]
vr2 = (y2.raw[,2] - y2.cali[,2])/y2.raw[,2]
hist(vr1, breaks = 20, main = "variance reduction", xlab = "exposure trait")
abline(v = mean(vr1), col = "blue")
abline(v = v, col = "red")

hist(vr2, breaks = 20, main = "variance reduction", xlab = "outcome trait")
abline(v = mean(vr2), col = "blue")
abline(v = v, col = "red")
```

### 3. Compute unadjusted SNP effect size & standard errors 

This is just to calculate GWAS summary stat from external data. 
```{r}
n = 1e3
N = 1e4

y1.ext <- matrix(NA, nrow = p, ncol = 2)
y2.ext <- matrix(NA, nrow = p, ncol = 2)
colnames(y1.ext) <- c("point_estimate", "std.error")
colnames(y2.ext) <- c("point_estimate", "std.error")

for (i in 1:p){
  dat1_ext <- cbind(dat[["pheno"]][,1], dat[["trans"]][, i]) 
  dat2_ext <- cbind(dat[["pheno"]][, 2], dat[["trans"]][, i]) 
  fit1 <- lm(dat1_ext[,1] ~ dat1_ext[,2])
  fit2 <- lm(dat2_ext[,1] ~ dat2_ext[,2])
  
  y1.ext[i, ] = c(summary(fit1)$coef[2,1], summary(fit1)$coef[2,2])
  y2.ext[i, ] = c(summary(fit2)$coef[2,1], summary(fit2)$coef[2,2])
}
```


Unadjusted internal data. 
```{r}
y1.int <- matrix(NA, nrow = p, ncol = 2)
y2.int <- matrix(NA, nrow = p, ncol = 2)

colnames(y1.int) <- c("point_estimate", "std.error")
colnames(y2.int) <- c("point_estimate", "std.error")

for (i in 1:p){
  dat1_int <- cbind(dat[["pheno"]][1:n,1], dat[["trans"]][1:n, i]) 
  dat2_int <- cbind(dat[["pheno"]][(1e3+1):(1e3+n), 2], dat[["trans"]][(1e3+1):(1e3+n), i]) 
  fit1 <- lm(dat1_int[,1] ~ dat1_int[,2])
  fit2 <- lm(dat2_int[,1] ~ dat2_int[,2])
  
  y1.int[i, ] = c(summary(fit1)$coef[2,1], summary(fit1)$coef[2,2])
  y2.int[i, ] = c(summary(fit2)$coef[2,1], summary(fit2)$coef[2,2])
}
```



### 4. Calculate z-scores 
```{r}
z2p <- function(zscore){
  pval <- pnorm(zscore, mean = 0, sd = 1, lower.tail = FALSE)*2
  return(pval)
}
```

```{r}
z1.cali = y1.cali[,1]/sqrt(y1.cali[,2])
z1.raw = y1.raw[,1]/sqrt(y1.raw[,2])
z2.cali = y2.cali[,1]/sqrt(y2.cali[,2])
z2.raw = y2.raw[,1]/sqrt(y2.raw[,2])

z1.ext = y1.ext[, 1]/y1.ext[,2]
z2.ext = y2.ext[, 1]/y2.ext[,2]
z1.int = y1.int[, 1]/y1.int[,2]
z2.int = y2.int[, 1]/y2.int[,2]

  
p1.cali = z2p(z1.cali)
p1.raw = z2p(z1.raw)
p2.cali = z2p(z2.cali)
p2.raw = z2p(z2.raw)

p1.ext = z2p(z1.ext)
p2.ext = z2p(z2.ext)
p1.int = z2p(z1.int)
p2.int = z2p(z2.int)
```

```{r}
par(mfrow = c(2,2))
plot(-log10(p1.cali), pch = 20, cex = 0.8, ylab = "-log10(pval)", main = "calibrated exposure")
plot(-log10(p2.cali), pch = 20, cex = 0.8, ylab = "-log10(pval)", main = "calibrated outcome")
plot(-log10(p1.raw), pch = 20, cex = 0.8, ylab = "-log10(pval)", main = "raw exposure")
plot(-log10(p2.raw), pch = 20, cex = 0.8, ylab = "-log10(pval)", main = "raw outcome")
```


```{r}
par(mfrow = c(2,2))
plot(-log10(p1.ext), pch = 20, cex = 0.8, ylab = "-log10(pval)", main = "unadjusted exposure: ext")
plot(-log10(p2.ext), pch = 20, cex = 0.8, ylab = "-log10(pval)", main = "unadjusted outcome: ext")
plot(-log10(p1.int), pch = 20, cex = 0.8, ylab = "-log10(pval)", main = "unadjusted exposure: int")
plot(-log10(p2.int), pch = 20, cex = 0.8, ylab = "-log10(pval)", main = "unadjusted outcome: int")
```


### 5. Perform different MR methods to assess the causal effect estimate.

```{r}
MRInput.cali <- mr_input(bx = y1.cali[,1],
                          bxse = sqrt(y1.cali[,2]),
                          by = y2.cali[,1],
                          byse = sqrt(y2.cali[,2]))

MRInput.raw <- mr_input(bx = y1.raw[,1],
                          bxse = sqrt(y1.raw[,2]),
                          by = y2.raw[,1],
                          byse = sqrt(y2.raw[,2]))


egger.cali <- mr_egger(MRInput.cali)
egger.raw <- mr_egger(MRInput.raw)

ivw.cali <- mr_ivw(MRInput.cali)
ivw.raw <- mr_ivw(MRInput.raw)
```

```{r}
egger.cali
egger.raw
ivw.cali
ivw.raw
```

```{r}

MRInput <- mr_input(bx = y1.ext[,1],
                          bxse = y1.ext[,2],
                          by = y2.ext[,1],
                          byse = y2.ext[,2])

egger.ext <- mr_egger(MRInput)
ivw.ext <- mr_ivw(MRInput)


MRInput <- mr_input(bx = y1.int[,1],
                          bxse = y1.int[,2],
                          by = y2.int[,1],
                          byse = y2.int[,2])

egger.int <- mr_egger(MRInput)
ivw.int <- mr_ivw(MRInput)

```

```{r}
egger.ext
ivw.ext
egger.int
ivw.int
```

### 5. Use mr.raps

#
```{r}
library(mr.raps)
```

```{r}
res.cali = mr.raps.mle(y1.cali[,1], y2.cali[,1], sqrt(y1.cali[,2]), sqrt(y2.cali[,2]))
res.raw = mr.raps.mle(y1.raw[,1], y2.raw[,1], sqrt(y1.raw[,2]), sqrt(y2.raw[,2]))
res.ext = mr.raps.mle(y1.ext[,1], y2.ext[,1], y1.ext[,2], y2.ext[,2])
res.int = mr.raps.mle(y1.int[,1], y2.int[,1], y1.int[,2], y2.int[,2])
```

```{r}
res.cali$beta.hat
res.raw$beta.hat
res.ext$beta.hat
res.int$beta.hat
```

```{r}
res.cali$beta.se
res.raw$beta.se
res.ext$beta.se
res.int$beta.se
```

