---
title: "UKB mr result: height as exposure"
author: "Yunqi Yang"
date: "8/6/2024"
output: html_document
---

#### SNP selection file available at:

https://github.com/yunqiyang0215/simulation-calibrated-mr/blob/master/analysis/snp_selection.R

```{r}
library(MendelianRandomization)
library(mr.raps)
```

```{r}
plot_combined_res <- function(res.mr, main = "IVW: BMI on diabetes",
                              colors = c("coral1", "skyblue", "gold"), pch = 20,
                              label1=c('1e-2', '1e-5', '1e-8'), 
                              label2 = c(1150, 564, 346), 
                              xlab = "P-value threshold", ylab = "Point estimate",
                              legend=c("Calibrated estimator", "Raw estimator", "External GWAS")){
  
  point_est_cali <- res.mr$cali[1, ]
  std_err_cali <- res.mr$cali[2, ]
  point_est_raw <- res.mr$raw[1, ]
  std_err_raw <- res.mr$raw[2, ]
  point_est_ext <- res.mr$ext[1, ]
  std_err_ext <- res.mr$ext[2, ]
  
  point_est_all <- c(point_est_cali, point_est_raw, point_est_ext)
  std_err_all <- c(std_err_cali, std_err_raw, std_err_ext)
  
  x_cali <- 1:3 - 0.1
  x_raw <- 1:3
  x_ext <- 1:3 + 0.1
  
  # Plotting
  plot(NULL, xlim=c(0.5, 3.5), ylim=c(min(min(point_est_all - 1.96 * std_err_all), 0)-0.2, max(point_est_all + 1.96 * std_err_all) +0.5), 
       xaxt='n', xlab=xlab, ylab=ylab, main= main)
  abline(h = 0, lty = 2, col = "grey")
  axis(1, at=1:3, labels=label1)
  mtext(label2, side = 1, at = 1:3, line = -1, cex = 0.6)
  
  # Plot cali data
  points(x_cali, point_est_cali, pch=pch, col = colors[1])
  arrows(x_cali, point_est_cali - 1.96 * std_err_cali, x_cali, point_est_cali + 1.96 * std_err_cali, angle=90, code=3, length = 0.05, lwd = 2, col=colors[1])
  
  # Plot raw data
  points(x_raw, point_est_raw, pch=pch, col=colors[2])
  arrows(x_raw, point_est_raw - 1.96 * std_err_raw, x_raw, point_est_raw + 1.96 * std_err_raw, angle=90, code=3, length = 0.05, lwd = 2, col=colors[2])
  
  # Plot ext data
  points(x_ext, point_est_ext, pch=pch, col=colors[3])
  arrows(x_ext, point_est_ext - 1.96 * std_err_ext, x_ext, point_est_ext + 1.96 * std_err_ext, angle=90, code=3, length = 0.05, lwd = 2, col=colors[3])
  
  # Add legend
  legend("topright", legend=legend, col=colors, pch=pch)
}
```

### 1. Trio: height on education years
```{r}
exp = "height"
out = "education_yrs"
```

```{r}
# exposure trait
dat.int1 <- list()
dat.ext1 <- list()

for (chr_num in 1:22) {
  f1 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/", exp, "_chr_", chr_num, ".txt")
  f2 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr", chr_num, ".", exp, ".glm.linear")
  
  int <- read.csv(f1, header = TRUE, sep = " ")
  ext <- read.csv(f2, header = TRUE, sep = "\t")
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 <- do.call(rbind, dat.int1)
dat.ext1 <- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = "variant", by.y = "ID", all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)
```

```{r}
# outcome trait
dat.int2 <- list()
dat.ext2 <- list()

for (chr_num in 1:22) {
  f1 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/", out, "_chr_", chr_num, ".txt")
  f2 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr", chr_num, ".", out, ".glm.linear")
  
  int <- read.csv(f1, header = TRUE, sep = " ")
  ext <- read.csv(f2, header = TRUE, sep = "\t")
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 <- do.call(rbind, dat.int2)
dat.ext2 <- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = "variant", by.y = "ID", all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)
```

```{r fig.width=10, fig.height = 4}
par(mfrow = c(1,3))
pvals = c(1e-2, 1e-5, 1e-8)

### IVW
res.mr = list(cali = matrix(NA, ncol = length(pvals), nrow = 2),
              raw = matrix(NA, ncol = length(pvals), nrow = 2),
              ext = matrix(NA, ncol = length(pvals), nrow = 2))
colnames(res.mr[['cali']]) = colnames(res.mr[['raw']]) = colnames(res.mr[['ext']]) = pvals
rownames(res.mr[['cali']]) = rownames(res.mr[['raw']]) = rownames(res.mr[['ext']]) = c("point_est", "std")

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0("/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_",
                                          pvals[i], ".txt"), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # calibrated estimator
  MRInput <- mr_input(bx = dat1$cali, bxse = sqrt(dat1$cali.var), by = dat2$cali, byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[["cali"]][,i] = c(mr@Estimate, mr@StdError)
 
  # raw estimator
  MRInput <- mr_input(bx = dat1$raw, bxse = sqrt(dat1$raw.var), by = dat2$raw, byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[["raw"]][,i] = c(mr@Estimate, mr@StdError)
  
  
  MRInput <- mr_input(bx = dat1$BETA, bxse = dat1$SE, by = dat2$BETA, byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[["ext"]][,i] = c(mr@Estimate, mr@StdError)
}

plot_combined_res(res.mr, main = paste0("IVW:", exp, " on ", out))


### mr.raps.mle
res.mr = list(cali = matrix(NA, ncol = length(pvals), nrow = 2),
              raw = matrix(NA, ncol = length(pvals), nrow = 2),
              ext = matrix(NA, ncol = length(pvals), nrow = 2))
colnames(res.mr[['cali']]) = colnames(res.mr[['raw']]) = colnames(res.mr[['ext']]) = pvals
rownames(res.mr[['cali']]) = rownames(res.mr[['raw']]) = rownames(res.mr[['ext']]) = c("point_est", "std")

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0("/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_",
                                          pvals[i], ".txt"), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # calibrated estimator
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["cali"]][,i] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["raw"]][,i] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["ext"]][,i] = c(point_est, stderr)
}

plot_combined_res(res.mr, main = paste0("mr.raps.mle:", exp, " on ", out))

# mr.raps.shrinkage
res.mr = list(cali = matrix(NA, ncol = length(pvals), nrow = 2),
              raw = matrix(NA, ncol = length(pvals), nrow = 2),
              ext = matrix(NA, ncol = length(pvals), nrow = 2))
colnames(res.mr[['cali']]) = colnames(res.mr[['raw']]) = colnames(res.mr[['ext']]) = pvals
rownames(res.mr[['cali']]) = rownames(res.mr[['raw']]) = rownames(res.mr[['ext']]) = c("point_est", "std")

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0("/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_",
                                          pvals[i], ".txt"), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  # calibrated estimator
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["cali"]][,i] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["raw"]][,i] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["ext"]][,i] = c(point_est, stderr)
}

plot_combined_res(res.mr, main = paste0("mr.raps.shrinkage:", exp, " on ", out))
```

### 2. Sibling: height on education years
```{r}
exp = "height"
out = "education_yrs"
```

```{r}
# exposure trait
dat.int1 <- list()
dat.ext1 <- list()

for (chr_num in 1:22) {
  f1 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/", exp, "_chr_", chr_num, ".txt")
  f2 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr", chr_num, ".", exp, ".glm.linear")
  
  int <- read.csv(f1, header = TRUE, sep = " ")
  ext <- read.csv(f2, header = TRUE, sep = "\t")
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 <- do.call(rbind, dat.int1)
dat.ext1 <- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = "variant", by.y = "ID", all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)
```

```{r}
# outcome trait
dat.int2 <- list()
dat.ext2 <- list()

for (chr_num in 1:22) {
  f1 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/", out, "_chr_", chr_num, ".txt")
  f2 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr", chr_num, ".", out, ".glm.linear")
  
  int <- read.csv(f1, header = TRUE, sep = " ")
  ext <- read.csv(f2, header = TRUE, sep = "\t")
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 <- do.call(rbind, dat.int2)
dat.ext2 <- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = "variant", by.y = "ID", all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)
```

```{r fig.width=10, fig.height = 4}
par(mfrow = c(1,3))
pvals = c(1e-2, 1e-5, 1e-8)

### IVW
res.mr = list(cali = matrix(NA, ncol = length(pvals), nrow = 2),
              raw = matrix(NA, ncol = length(pvals), nrow = 2),
              ext = matrix(NA, ncol = length(pvals), nrow = 2))
colnames(res.mr[['cali']]) = colnames(res.mr[['raw']]) = colnames(res.mr[['ext']]) = pvals
rownames(res.mr[['cali']]) = rownames(res.mr[['raw']]) = rownames(res.mr[['ext']]) = c("point_est", "std")

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0("/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_",
                                          pvals[i], ".txt"), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # calibrated estimator
  MRInput <- mr_input(bx = dat1$cali, bxse = sqrt(dat1$cali.var), by = dat2$cali, byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[["cali"]][,i] = c(mr@Estimate, mr@StdError)
 
  # raw estimator
  MRInput <- mr_input(bx = dat1$raw, bxse = sqrt(dat1$raw.var), by = dat2$raw, byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[["raw"]][,i] = c(mr@Estimate, mr@StdError)
  
  
  MRInput <- mr_input(bx = dat1$BETA, bxse = dat1$SE, by = dat2$BETA, byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[["ext"]][,i] = c(mr@Estimate, mr@StdError)
}

plot_combined_res(res.mr, main = paste0("IVW:", exp, " on ", out))


### mr.raps.mle
res.mr = list(cali = matrix(NA, ncol = length(pvals), nrow = 2),
              raw = matrix(NA, ncol = length(pvals), nrow = 2),
              ext = matrix(NA, ncol = length(pvals), nrow = 2))
colnames(res.mr[['cali']]) = colnames(res.mr[['raw']]) = colnames(res.mr[['ext']]) = pvals
rownames(res.mr[['cali']]) = rownames(res.mr[['raw']]) = rownames(res.mr[['ext']]) = c("point_est", "std")

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0("/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_",
                                          pvals[i], ".txt"), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # calibrated estimator
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["cali"]][,i] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["raw"]][,i] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["ext"]][,i] = c(point_est, stderr)
}

plot_combined_res(res.mr, main = paste0("mr.raps.mle:", exp, " on ", out))

# mr.raps.shrinkage
res.mr = list(cali = matrix(NA, ncol = length(pvals), nrow = 2),
              raw = matrix(NA, ncol = length(pvals), nrow = 2),
              ext = matrix(NA, ncol = length(pvals), nrow = 2))
colnames(res.mr[['cali']]) = colnames(res.mr[['raw']]) = colnames(res.mr[['ext']]) = pvals
rownames(res.mr[['cali']]) = rownames(res.mr[['raw']]) = rownames(res.mr[['ext']]) = c("point_est", "std")

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0("/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_",
                                          pvals[i], ".txt"), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  # calibrated estimator
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["cali"]][,i] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["raw"]][,i] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["ext"]][,i] = c(point_est, stderr)
}

plot_combined_res(res.mr, main = paste0("mr.raps.shrinkage:", exp, " on ", out))
```


### 3. Trio: height on smoking
```{r}
exp = "height"
out = "smoking"
```

```{r}
# exposure trait
dat.int1 <- list()
dat.ext1 <- list()

for (chr_num in 1:22) {
  f1 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/", exp, "_chr_", chr_num, ".txt")
  f2 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr", chr_num, ".", exp, ".glm.linear")
  
  int <- read.csv(f1, header = TRUE, sep = " ")
  ext <- read.csv(f2, header = TRUE, sep = "\t")
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 <- do.call(rbind, dat.int1)
dat.ext1 <- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = "variant", by.y = "ID", all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)
```

```{r}
# outcome trait
dat.int2 <- list()
dat.ext2 <- list()

for (chr_num in 1:22) {
  f1 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_trio/", out, "_chr_", chr_num, ".txt")
  f2 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr", chr_num, ".", out, ".glm.linear")
  
  int <- read.csv(f1, header = TRUE, sep = " ")
  ext <- read.csv(f2, header = TRUE, sep = "\t")
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 <- do.call(rbind, dat.int2)
dat.ext2 <- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = "variant", by.y = "ID", all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)
```

```{r fig.width=10, fig.height = 4}
par(mfrow = c(1,3))
pvals = c(1e-2, 1e-5, 1e-8)

### IVW
res.mr = list(cali = matrix(NA, ncol = length(pvals), nrow = 2),
              raw = matrix(NA, ncol = length(pvals), nrow = 2),
              ext = matrix(NA, ncol = length(pvals), nrow = 2))
colnames(res.mr[['cali']]) = colnames(res.mr[['raw']]) = colnames(res.mr[['ext']]) = pvals
rownames(res.mr[['cali']]) = rownames(res.mr[['raw']]) = rownames(res.mr[['ext']]) = c("point_est", "std")

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0("/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_",
                                          pvals[i], ".txt"), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # calibrated estimator
  MRInput <- mr_input(bx = dat1$cali, bxse = sqrt(dat1$cali.var), by = dat2$cali, byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[["cali"]][,i] = c(mr@Estimate, mr@StdError)
 
  # raw estimator
  MRInput <- mr_input(bx = dat1$raw, bxse = sqrt(dat1$raw.var), by = dat2$raw, byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[["raw"]][,i] = c(mr@Estimate, mr@StdError)
  
  
  MRInput <- mr_input(bx = dat1$BETA, bxse = dat1$SE, by = dat2$BETA, byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[["ext"]][,i] = c(mr@Estimate, mr@StdError)
}

plot_combined_res(res.mr, main = paste0("IVW:", exp, " on ", out))


### mr.raps.mle
res.mr = list(cali = matrix(NA, ncol = length(pvals), nrow = 2),
              raw = matrix(NA, ncol = length(pvals), nrow = 2),
              ext = matrix(NA, ncol = length(pvals), nrow = 2))
colnames(res.mr[['cali']]) = colnames(res.mr[['raw']]) = colnames(res.mr[['ext']]) = pvals
rownames(res.mr[['cali']]) = rownames(res.mr[['raw']]) = rownames(res.mr[['ext']]) = c("point_est", "std")

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0("/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_",
                                          pvals[i], ".txt"), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # calibrated estimator
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["cali"]][,i] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["raw"]][,i] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["ext"]][,i] = c(point_est, stderr)
}

plot_combined_res(res.mr, main = paste0("mr.raps.mle:", exp, " on ", out))

# mr.raps.shrinkage
res.mr = list(cali = matrix(NA, ncol = length(pvals), nrow = 2),
              raw = matrix(NA, ncol = length(pvals), nrow = 2),
              ext = matrix(NA, ncol = length(pvals), nrow = 2))
colnames(res.mr[['cali']]) = colnames(res.mr[['raw']]) = colnames(res.mr[['ext']]) = pvals
rownames(res.mr[['cali']]) = rownames(res.mr[['raw']]) = rownames(res.mr[['ext']]) = c("point_est", "std")

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0("/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_",
                                          pvals[i], ".txt"), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  # calibrated estimator
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["cali"]][,i] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["raw"]][,i] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["ext"]][,i] = c(point_est, stderr)
}

plot_combined_res(res.mr, main = paste0("mr.raps.shrinkage:", exp, " on ", out))
```

### 4. Sibling: height on smoking
```{r}
exp = "height"
out = "smoking"
```

```{r}
# exposure trait
dat.int1 <- list()
dat.ext1 <- list()

for (chr_num in 1:22) {
  f1 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/", exp, "_chr_", chr_num, ".txt")
  f2 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr", chr_num, ".", exp, ".glm.linear")
  
  int <- read.csv(f1, header = TRUE, sep = " ")
  ext <- read.csv(f2, header = TRUE, sep = "\t")
  dat.int1[[chr_num]] = int
  dat.ext1[[chr_num]] = ext
}

dat.int1 <- do.call(rbind, dat.int1)
dat.ext1 <- do.call(rbind, dat.ext1)

dat.exp = merge(dat.int1, dat.ext1, by.x = "variant", by.y = "ID", all = FALSE)
# flip sign of effect if the allele tested is different
dat.exp$BETA = ifelse(dat.exp$allele.test == dat.exp$A1, dat.exp$BETA, -dat.exp$BETA)
```

```{r}
# outcome trait
dat.int2 <- list()
dat.ext2 <- list()

for (chr_num in 1:22) {
  f1 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/mr_gwas_sibling/", out, "_chr_", chr_num, ".txt")
  f2 <- paste0("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/202407/chr", chr_num, ".", out, ".glm.linear")
  
  int <- read.csv(f1, header = TRUE, sep = " ")
  ext <- read.csv(f2, header = TRUE, sep = "\t")
  dat.int2[[chr_num]] = int
  dat.ext2[[chr_num]] = ext
}

dat.int2 <- do.call(rbind, dat.int2)
dat.ext2 <- do.call(rbind, dat.ext2)

dat.out = merge(dat.int2, dat.ext2, by.x = "variant", by.y = "ID", all = FALSE)
# flip sign of effect if the allele tested is different
dat.out$BETA = ifelse(dat.out$allele.test == dat.out$A1, dat.out$BETA, -dat.out$BETA)
```

```{r fig.width=10, fig.height = 4}
par(mfrow = c(1,3))
pvals = c(1e-2, 1e-5, 1e-8)

### IVW
res.mr = list(cali = matrix(NA, ncol = length(pvals), nrow = 2),
              raw = matrix(NA, ncol = length(pvals), nrow = 2),
              ext = matrix(NA, ncol = length(pvals), nrow = 2))
colnames(res.mr[['cali']]) = colnames(res.mr[['raw']]) = colnames(res.mr[['ext']]) = pvals
rownames(res.mr[['cali']]) = rownames(res.mr[['raw']]) = rownames(res.mr[['ext']]) = c("point_est", "std")

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0("/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_",
                                          pvals[i], ".txt"), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # calibrated estimator
  MRInput <- mr_input(bx = dat1$cali, bxse = sqrt(dat1$cali.var), by = dat2$cali, byse = sqrt(dat2$cali.var))
  mr = mr_ivw(MRInput)
  res.mr[["cali"]][,i] = c(mr@Estimate, mr@StdError)
 
  # raw estimator
  MRInput <- mr_input(bx = dat1$raw, bxse = sqrt(dat1$raw.var), by = dat2$raw, byse = sqrt(dat2$raw.var))
  mr = mr_ivw(MRInput)
  res.mr[["raw"]][,i] = c(mr@Estimate, mr@StdError)
  
  
  MRInput <- mr_input(bx = dat1$BETA, bxse = dat1$SE, by = dat2$BETA, byse = dat1$SE)
  mr = mr_ivw(MRInput)
  res.mr[["ext"]][,i] = c(mr@Estimate, mr@StdError)
}

plot_combined_res(res.mr, main = paste0("IVW:", exp, " on ", out))


### mr.raps.mle
res.mr = list(cali = matrix(NA, ncol = length(pvals), nrow = 2),
              raw = matrix(NA, ncol = length(pvals), nrow = 2),
              ext = matrix(NA, ncol = length(pvals), nrow = 2))
colnames(res.mr[['cali']]) = colnames(res.mr[['raw']]) = colnames(res.mr[['ext']]) = pvals
rownames(res.mr[['cali']]) = rownames(res.mr[['raw']]) = rownames(res.mr[['ext']]) = c("point_est", "std")

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0("/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_",
                                          pvals[i], ".txt"), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  sum(dat1$variant != dat2$variant)
  
  # calibrated estimator
  mr = mr.raps.mle(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["cali"]][,i] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.mle(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["raw"]][,i] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.mle(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["ext"]][,i] = c(point_est, stderr)
}

plot_combined_res(res.mr, main = paste0("mr.raps.mle:", exp, " on ", out))

# mr.raps.shrinkage
res.mr = list(cali = matrix(NA, ncol = length(pvals), nrow = 2),
              raw = matrix(NA, ncol = length(pvals), nrow = 2),
              ext = matrix(NA, ncol = length(pvals), nrow = 2))
colnames(res.mr[['cali']]) = colnames(res.mr[['raw']]) = colnames(res.mr[['ext']]) = pvals
rownames(res.mr[['cali']]) = rownames(res.mr[['raw']]) = rownames(res.mr[['ext']]) = c("point_est", "std")

for (i in 1:length(pvals)){
  bmi_snps = read.table(paste0("/Users/nicholeyang/Downloads/simulation-calibrated-mr/data/snp_list/height_",
                                          pvals[i], ".txt"), header = FALSE)
  dat1 = dat.exp[dat.exp$variant %in% bmi_snps$V1, ]
  dat2 = dat.out[dat.out$variant %in% bmi_snps$V1, ]
  # calibrated estimator
  mr = mr.raps.shrinkage(dat1$cali, dat2$cali, sqrt(dat1$cali.var), sqrt(dat2$cali.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["cali"]][,i] = c(point_est, stderr)
  
  # adj+uncalibrated
  mr = mr.raps.shrinkage(dat1$raw, dat2$raw, sqrt(dat1$raw.var), sqrt(dat2$raw.var))
  
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["raw"]][,i] = c(point_est, stderr)
  
  # external gwas
  mr = mr.raps.shrinkage(dat1$BETA, dat2$BETA, dat1$SE, dat2$SE)
  point_est = mr$beta.hat
  stderr = mr$beta.se
  res.mr[["ext"]][,i] = c(point_est, stderr)
}

plot_combined_res(res.mr, main = paste0("mr.raps.shrinkage:", exp, " on ", out))
```
