---
title: "Simulate data for MR using parental phenotype"
author: "Yunqi Yang"
date: "7/28/2023"
output: html_document
---


### Description: 

Here I describe the steps for doing simulation using minor allele frequency (MAF) from real genotype data from UKB. The MAF is calculated using this file: /project2/mstephens/yunqiyang/calibrated_mr/sim_realdata/geno_individual.R.


**Step 1**: Simulate **parents** genotype based on real MAF. Number of SNPs $p = 100$, sample size = $2*N$.

**Step 2**: Create **complete offspring data** with sample size $N$. The complete data is data with both transmitted $T$ and non-transmitted ($NT$) alleles. The external data will be of size $N$ but only have transmitted alleles. The internal data will be of size $n$ and with both $T$ and $NT$. 

**Step 3**: Simulate $Y_1$ and $Y_2$, exposure trait and outcome trait. We use the following model to simulate by first simulating parental phenotypes. 

**Parents phenotype**

$G$ parents genotype matrix, size $2N\times p$. 

$$
\begin{split}
b^p&\sim N(0, 1),\quad \epsilon\sim N(0, 1)\\
Y^p &= G^pb^p+\epsilon
\end{split}
$$

**Children exposure & outcome traits**

$\beta$ denotes the causal effect. $T$ denotes transmitted allelle matrix, and $NT$ denotes non-transmitted allele matrix. 
$$
\begin{split}
\beta & =1\\
b&\sim N(0, 1), \quad \epsilon\sim N(0, 1)\\
Y_1 &= Tb + r_1(Y^m+Y^f) + \epsilon\\
Y_2 &= \beta Y_1 + r_2(Y^m + Y^f) + \epsilon
\end{split}
$$

```{r}
### Helper functions to simulate trio data. 
# @par.allele: allele value of a parent
sample_allele <- function(par.allele){
  if (par.allele == 0) val = 0
  if (par.allele == 2) val = 1
  if (par.allele == 1) val = sample(c(0, 1), 1)
  return(val)
}

# @par1: genotyp of parent 1
# @par2: genotype of parent 2
# @return: transmitted and non-transmitted alleles of offspring
sim_offspring <- function(par1, par2){
  p = length(par1)
  offspring.t = rep(NA, p)
  offspring.nt = rep(NA, p)
  for (i in 1:p){
    a1 <- sample_allele(par1[i])
    a2 <- sample_allele(par2[i])
    t <- a1 + a2
    nt <- par1[i] + par2[i] - t
    offspring.t[i] <- t
    offspring.nt[i] <- nt
  }
  return(list(offspring.t = offspring.t, offspring.nt = offspring.nt))
}
```


```{r}
geno_real <- readRDS("./data/geno_individual.rds")
maf <- colMeans(geno_real)/2
set.seed(1)
N = 1e4  
p = 100

# 1. Simulate parents genotype based on true MAF. 
# Independent bernoulli trials. That is, the two alleles of parents are independent.
geno = matrix(NA, ncol = p, nrow = 2*N)
for (i in 1:p){
  geno[,i] <- rbinom(2*N, size = 2, prob = maf[i])
}

# 2. Simulate Y_pa: parents phenotype
b.pa = rnorm(p)
Y.pa = geno %*% b.pa + rnorm(2*N, 0, 1)
```

```{r}
# 3. Simulate offspring genotype data.
# N by p matrix
trans = matrix(NA, ncol = p, nrow = N)
nontrans = matrix(NA, ncol = p, nrow = N)
for (i in 1:N){
  par1 = geno[2*i-1, ]
  par2 = geno[2*i, ]
  offspring <- sim_offspring(par1, par2)
  trans[i, ] <- offspring$offspring.t
  nontrans[i, ] <-  offspring$offspring.nt
}


# 4. Simulate trait
set.seed(1)
beta = 1 # causal effect
r1 = 0.1
r2 = 0.5
b = rnorm(p)
Y1 = rep(NA, N)
Y2 = rep(NA, N)
for (i in 1:N){
  pa_effect = Y.pa[2*i-1] + Y.pa[2*i]
  Y1[i] = sum(trans[i, ] * b) + r1*pa_effect + rnorm(1)
  Y2[i] = beta*Y1[i] + r2*pa_effect + rnorm(1)
}
pheno <- cbind(Y1, Y2)
dat = list(trans = trans, nontrans = nontrans, pheno = pheno)
```
