---
title: "Calibration on ukb trio data"
author: "Yunqi Yang"
date: "5/8/2024"
output: html_document
---

### 1. BMI

```{r}
sumstat = readRDS("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/trio_calibration_bmi.rds")
ext = read.csv("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/gwas_bmi.linear", sep = "\t")
```

```{r}
sumstat = as.data.frame(sumstat)
sumstat[, c(3:8)] = apply(sumstat[, c(3:8)], 2, as.numeric)
dat = merge(sumstat, ext, by.x = "variant", by.y = "ID", all = FALSE)
# flip sign of effect if the allele tested is different
dat$BETA = ifelse(dat$allele.test == dat$A1, dat$BETA, -dat$BETA)
```

```{r fig.width=8, fig.height=8}
# check estimate scale
par(mfrow = c(2,2))

plot(dat$cali, dat$raw, xlab = "calibrated estimator", ylab = "uncalibrated estimator")
abline(a = 0, b = 1, col = "red")
plot(dat$BETA, dat$int, xlab = "external gwas", ylab = "internal gwas")
abline(a = 0, b = 1, col = "red")

hist(dat$cali - dat$raw, main = "Differece: calibrated estimator - uncalibrated estimator", xlab = "")
hist(dat$int - dat$BETA, main = "Difference: internal gwas - external gwas", xlab = "")
cor(dat$cali, dat$raw)
cor(dat$BETA, dat$int)
```

```{r}
fit = lm(dat$int ~ dat$BETA)
summary(fit)
```

```{r }
pval.ext = pnorm(abs(dat$BETA/dat$SE), lower.tail = FALSE)
pval.cali = pnorm(abs(dat$cali/sqrt(dat$cali.var)), lower.tail = FALSE)
pval.uncali = pnorm(abs(dat$raw/sqrt(dat$raw.var)), lower.tail = FALSE)

sum(pval.ext < 1e-8)
sum(pval.cali < 1e-3)
sum(pval.uncali < 1e-3)
```

#### Calibration result

```{r fig.width=10, fig.height=6}
par(mfrow = c(1, 2))
boxplot(dat[, c("cali.var", "raw.var")], ylab = "variance")
vr = dat$cali.var/dat$raw.var
hist(vr, main = "variance ratio", xlab = "")
```

```{r}
dat[vr < 0.2, c("variant", "allele.test", "cali.var", "raw.var")]
```






### 2. DBP

```{r}
sumstat = readRDS("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/trio_calibration_dbp.rds")
ext = read.csv("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/gwas_dbp.linear", sep = "\t")
```

```{r}
sumstat = as.data.frame(sumstat)
sumstat[, c(3:8)] = apply(sumstat[, c(3:8)], 2, as.numeric)
dat = merge(sumstat, ext, by.x = "variant", by.y = "ID", all = FALSE)
# flip sign of effect if the allele tested is different
dat$BETA = ifelse(dat$allele.test == dat$A1, dat$BETA, -dat$BETA)
```

```{r fig.width=8, fig.height=8}
# check estimate scale
par(mfrow = c(2,2))

plot(dat$cali, dat$raw, xlab = "calibrated estimator", ylab = "uncalibrated estimator")
abline(a = 0, b = 1, col = "red")
plot(dat$BETA, dat$int, xlab = "external gwas", ylab = "internal gwas")
abline(a = 0, b = 1, col = "red")

hist(dat$cali - dat$raw, main = "Differece: calibrated estimator - uncalibrated estimator", xlab = "")
hist(dat$int - dat$BETA, main = "Difference: internal gwas - external gwas", xlab = "")
cor(dat$cali, dat$raw)
cor(dat$BETA, dat$int)
```

```{r}
fit = lm(dat$int ~ dat$BETA)
summary(fit)
```

```{r }
pval.ext = pnorm(abs(dat$BETA/dat$SE), lower.tail = FALSE)
pval.cali = pnorm(abs(dat$cali/sqrt(dat$cali.var)), lower.tail = FALSE)
pval.uncali = pnorm(abs(dat$raw/sqrt(dat$raw.var)), lower.tail = FALSE)

sum(pval.ext < 1e-8)
sum(pval.cali < 1e-3)
sum(pval.uncali < 1e-3)
```

#### Calibration result

```{r fig.width=10, fig.height=6}
par(mfrow = c(1, 2))
boxplot(dat[, c("cali.var", "raw.var")], ylab = "variance")
vr = dat$cali.var/dat$raw.var
hist(vr, main = "variance ratio", xlab = "")
```

```{r}
dat[vr < 0.2, c("variant", "allele.test", "cali.var", "raw.var")]
```


### 3. SBP

```{r}
sumstat = readRDS("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/trio_calibration_sbp.rds")
ext = read.csv("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/gwas_sbp.linear", sep = "\t")
```

```{r}
sumstat = as.data.frame(sumstat)
sumstat[, c(3:8)] = apply(sumstat[, c(3:8)], 2, as.numeric)
dat = merge(sumstat, ext, by.x = "variant", by.y = "ID", all = FALSE)
# flip sign of effect if the allele tested is different
dat$BETA = ifelse(dat$allele.test == dat$A1, dat$BETA, -dat$BETA)
```

```{r fig.width=8, fig.height=8}
# check estimate scale
par(mfrow = c(2,2))

plot(dat$cali, dat$raw, xlab = "calibrated estimator", ylab = "uncalibrated estimator")
abline(a = 0, b = 1, col = "red")
plot(dat$BETA, dat$int, xlab = "external gwas", ylab = "internal gwas")
abline(a = 0, b = 1, col = "red")

hist(dat$cali - dat$raw, main = "Differece: calibrated estimator - uncalibrated estimator", xlab = "")
hist(dat$int - dat$BETA, main = "Difference: internal gwas - external gwas", xlab = "")
cor(dat$cali, dat$raw)
cor(dat$BETA, dat$int)
```

```{r}
fit = lm(dat$int ~ dat$BETA)
summary(fit)
```

```{r }
pval.ext = pnorm(abs(dat$BETA/dat$SE), lower.tail = FALSE)
pval.cali = pnorm(abs(dat$cali/sqrt(dat$cali.var)), lower.tail = FALSE)
pval.uncali = pnorm(abs(dat$raw/sqrt(dat$raw.var)), lower.tail = FALSE)

sum(pval.ext < 1e-8)
sum(pval.cali < 1e-3)
sum(pval.uncali < 1e-3)
```

#### Calibration result

```{r fig.width=10, fig.height=6}
par(mfrow = c(1, 2))
boxplot(dat[, c("cali.var", "raw.var")], ylab = "variance")
vr = dat$cali.var/dat$raw.var
hist(vr, main = "variance ratio", xlab = "")
```

```{r}
dat[vr < 0.2, c("variant", "allele.test", "cali.var", "raw.var")]
```


### 4. diabetes

```{r}
sumstat = readRDS("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/trio_calibration_diabetes.rds")
ext = read.csv("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/gwas_diabetes.linear", sep = "\t")
```

```{r}
sumstat = as.data.frame(sumstat)
sumstat[, c(3:8)] = apply(sumstat[, c(3:8)], 2, as.numeric)
dat = merge(sumstat, ext, by.x = "variant", by.y = "ID", all = FALSE)
# flip sign of effect if the allele tested is different
dat$BETA = ifelse(dat$allele.test == dat$A1, dat$BETA, -dat$BETA)
```

```{r fig.width=8, fig.height=8}
# check estimate scale
par(mfrow = c(2,2))

plot(dat$cali, dat$raw, xlab = "calibrated estimator", ylab = "uncalibrated estimator")
abline(a = 0, b = 1, col = "red")
plot(dat$BETA, dat$int, xlab = "external gwas", ylab = "internal gwas")
abline(a = 0, b = 1, col = "red")

hist(dat$cali - dat$raw, main = "Differece: calibrated estimator - uncalibrated estimator", xlab = "")
hist(dat$int - dat$BETA, main = "Difference: internal gwas - external gwas", xlab = "")
cor(dat$cali, dat$raw)
cor(dat$BETA, dat$int)
```

```{r}
fit = lm(dat$int ~ dat$BETA)
summary(fit)
```

```{r }
pval.ext = pnorm(abs(dat$BETA/dat$SE), lower.tail = FALSE)
pval.cali = pnorm(abs(dat$cali/sqrt(dat$cali.var)), lower.tail = FALSE)
pval.uncali = pnorm(abs(dat$raw/sqrt(dat$raw.var)), lower.tail = FALSE)

sum(pval.ext < 1e-8)
sum(pval.cali < 1e-3)
sum(pval.uncali < 1e-3)
```


#### Calibration result

```{r fig.width=10, fig.height=6}
par(mfrow = c(1, 2))
boxplot(dat[, c("cali.var", "raw.var")], ylab = "variance")
vr = dat$cali.var/dat$raw.var
hist(vr, main = "variance ratio", xlab = "")
```

```{r}
dat[vr < 0.2, c("variant", "allele.test", "cali.var", "raw.var")]
```

```{r}
dat[vr > 0.95, c("variant", "allele.test", "cali.var", "raw.var")]
```

### 5. Education years  

```{r}
sumstat = readRDS("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/trio_calibration_education_yrs.rds")
ext = read.csv("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/gwas_education_yrs.linear", sep = "\t")
```

```{r}
sumstat = as.data.frame(sumstat)
sumstat[, c(3:8)] = apply(sumstat[, c(3:8)], 2, as.numeric)
dat = merge(sumstat, ext, by.x = "variant", by.y = "ID", all = FALSE)
# flip sign of effect if the allele tested is different
dat$BETA = ifelse(dat$allele.test == dat$A1, dat$BETA, -dat$BETA)
```

```{r fig.width=8, fig.height=8}
# check estimate scale
par(mfrow = c(2,2))

plot(dat$cali, dat$raw, xlab = "calibrated estimator", ylab = "uncalibrated estimator")
abline(a = 0, b = 1, col = "red")
plot(dat$BETA, dat$int, xlab = "external gwas", ylab = "internal gwas")
abline(a = 0, b = 1, col = "red")

hist(dat$cali - dat$raw, main = "Differece: calibrated estimator - uncalibrated estimator", xlab = "")
hist(dat$int - dat$BETA, main = "Difference: internal gwas - external gwas", xlab = "")
cor(dat$cali, dat$raw)
cor(dat$BETA, dat$int)
```

```{r}
fit = lm(dat$int ~ dat$BETA)
summary(fit)
```

```{r }
pval.ext = pnorm(abs(dat$BETA/dat$SE), lower.tail = FALSE)
pval.cali = pnorm(abs(dat$cali/sqrt(dat$cali.var)), lower.tail = FALSE)
pval.uncali = pnorm(abs(dat$raw/sqrt(dat$raw.var)), lower.tail = FALSE)

sum(pval.ext < 1e-8)
sum(pval.cali < 1e-3)
sum(pval.uncali < 1e-3)
```

#### Calibration result

```{r fig.width=10, fig.height=6}
par(mfrow = c(1, 2))
boxplot(dat[, c("cali.var", "raw.var")], ylab = "variance")
vr = dat$cali.var/dat$raw.var
hist(vr, main = "variance ratio", xlab = "")
```

```{r}
dat[vr < 0.2, c("variant", "allele.test", "cali.var", "raw.var")]
```



### Summary
```{r}
bmi = readRDS("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/trio_calibration_bmi.rds")
sbp = readRDS("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/trio_calibration_sbp.rds")
dbp = readRDS("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/trio_calibration_dbp.rds")
diabetes = readRDS("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/trio_calibration_diabetes.rds")
edu = readRDS("/Users/nicholeyang/Downloads/calibrated_estimator/real-data-analysis/result/trio_calibration_education_yrs.rds")
```

```{r}
dat = list(bmi, sbp, dbp, diabetes, edu)
for (i in 1:length(dat)){
  dat[[i]] = as.data.frame(dat[[i]])
  dat[[i]][, c(3:8)] = apply(dat[[i]][, c(3:8)], 2, as.numeric)
}

```

```{r}
diff = sapply(1:5, function(i) dat[[i]]$cali - dat[[i]]$raw)
vr =  sapply(1:5, function(i) 1 - dat[[i]]$cali.var/dat[[i]]$raw.var)
colnames(diff) = c("bmi", "sbp", "dbp", "diabetes", "education years")
colnames(vr) = c("bmi", "sbp", "dbp", "diabetes", "education years")
```

```{r fig.width = 10, fig.height=5}
par(mfrow = c(1,2))
boxplot(diff, main = "Difference beween calibrated estimator and raw estimator", 
        ylab = "", names = colnames(diff), pch = 20, cex = 0.6)
boxplot(vr, main = "variance reduction",names = colnames(diff), pch = 20, cex = 0.6)
```



